diff --git a/work/cmplrflags.mk b/work/cmplrflags.mk
index aa57644..5f65b1f 100644
--- a/work/cmplrflags.mk
+++ b/work/cmplrflags.mk
@@ -166,11 +166,6 @@ ifeq ($(compiler),gfortran)
         FFLAGS2 := $(FFLAGS2) -L/usr/lib/x86_64-linux-gnu
         FFLAGS3 := $(FFLAGS3) -L/usr/lib/x86_64-linux-gnu
      endif
-     ifeq ($(MACHINENAME),rostam)
-        FFLAGS1 := $(FFLAGS1) -I/usr/lib64/gfortran/modules
-        FFLAGS2 := $(FFLAGS2) -I/usr/lib64/gfortran/modules
-        FFLAGS3 := $(FFLAGS3) -I/usr/lib64/gfortran/modules
-     endif
      FLIBS      := $(FLIBS) -lnetcdff
   endif
   IMODS 	:=  -I
@@ -274,7 +269,7 @@ ifeq ($(compiler),intel)
         CFLAGS  := $(INCDIRS) -g -O0 -traceback -DLINUX -xSSE4.2
         FLIBS   := $(INCDIRS) -xSSE4.2
      endif
-  ifeq ($(MACHINENAME),supermic) 
+  ifeq ($(MACHINENAME),supermic)
      FFLAGS1 := $(INCDIRS) -O3 -FI -assume byterecl -132 -xAVX -assume buffered_io
      CFLAGS  := $(INCDIRS) -O3 -DLINUX -xAVX
      FLIBS   := $(INCDIRS) -xAVX
@@ -284,16 +279,7 @@ ifeq ($(compiler),intel)
         FLIBS   := $(INCDIRS) -xAVX
      endif
   endif
-  ifeq ($(MACHINENAME),queenbeeC) 
-     FFLAGS1 := $(INCDIRS) -O3 -FI -assume byterecl -132 -xCORE-AVX512 -assume buffered_io
-     CFLAGS  := $(INCDIRS) -O3 -DLINUX -xCORE-AVX512 
-     FLIBS   := $(INCDIRS) -xCORE-AVX512 
-     ifeq ($(DEBUG),trace)
-        FFLAGS1 := $(INCDIRS) -g -O0 -traceback -FI -assume byterecl -132 -xCORE-AVX512 -assume buffered_io
-        CFLAGS  := $(INCDIRS) -g -O0 -traceback -DLINUX -xCORE-AVX512 
-        FLIBS   := $(INCDIRS) -xCORE-AVX512 
-     endif
-  ifeq ($(MACHINENAME),supermic) 
+  ifeq ($(MACHINENAME),rostam)
      FFLAGS1 := $(INCDIRS) -O3 -FI -assume byterecl -132 -xAVX -assume buffered_io
      CFLAGS  := $(INCDIRS) -O3 -DLINUX -xAVX
      FLIBS   := $(INCDIRS) -xAVX
@@ -303,6 +289,15 @@ ifeq ($(compiler),intel)
         FLIBS   := $(INCDIRS) -xAVX
      endif
   endif
+  ifeq ($(MACHINENAME),queenbeeC)
+     FFLAGS1 := $(INCDIRS) -O3 -FI -assume byterecl -132 -xCORE-AVX512 -assume buffered_io
+     CFLAGS  := $(INCDIRS) -O3 -DLINUX -xCORE-AVX512
+     FLIBS   := $(INCDIRS) -xCORE-AVX512
+     ifeq ($(DEBUG),trace)
+        FFLAGS1 := $(INCDIRS) -g -O0 -traceback -FI -assume byterecl -132 -xCORE-AVX512 -assume buffered_io
+        CFLAGS  := $(INCDIRS) -g -O0 -traceback -DLINUX -xCORE-AVX512
+        FLIBS   := $(INCDIRS) -xCORE-AVX512
+     endif
   endif
   endif
   #
@@ -338,6 +333,9 @@ ifeq ($(compiler),intel)
      ifeq ($(MACHINENAME),supermic)
         FLIBS      := $(FLIBS) -L$(NETCDFHOME)/lib -lnetcdff -L$(NETCDFHOME)/lib -lnetcdf -lnetcdf
      endif
+     ifeq ($(MACHINENAME),rostam)
+        FLIBS      := $(FLIBS) -L$(NETCDFHOME)/lib -lnetcdff -L$(NETCDFHOME)/lib -lnetcdf -lnetcdf
+     endif
      ifeq ($(MACHINENAME),stampede)
         FLIBS      := $(FLIBS) -L$(NETCDFHOME)/lib -lnetcdff -lnetcdf
      endif
