diff --git a/thirdparty/swan/macros.inc.kraken b/thirdparty/swan/macros.inc.kraken
index a8cf30b..7245699 100644
--- a/thirdparty/swan/macros.inc.kraken
+++ b/thirdparty/swan/macros.inc.kraken
@@ -7,9 +7,9 @@ F90_OMP = ftn
 F90_MPI = ftn 
 FLAGS_OPT =
 FLAGS_MSC = 
-FLAGS_SER = -O3 -static 
+FLAGS_SER = -O2 -static 
 FLAGS_OMP = -mp
-FLAGS_MPI = -O3 -static 
+FLAGS_MPI = -O2 -static 
 INCS_SER =
 INCS_OMP =
 INCS_MPI =
diff --git a/thirdparty/swan/macros.inc.queenbee b/thirdparty/swan/macros.inc.queenbee
index e35a794..cfa6223 100644
--- a/thirdparty/swan/macros.inc.queenbee
+++ b/thirdparty/swan/macros.inc.queenbee
@@ -5,7 +5,7 @@ F90_SER = ifort
 F90_OMP = ifort
 F90_MPI = mpif90
 FLAGS_OPT = -O2
-FLAGS_MSC = -W0 -assume byterecl -traceback -diag-disable 8290 -diag-disable 8291 -diag-disable 8293 -assume buffered_io -xSSE4.2
+FLAGS_MSC = -W0 -assume byterecl -traceback -diag-disable 8290 -diag-disable 8291 -diag-disable 8293 -xSSE4.2
 FLAGS90_MSC = $(FLAGS_MSC)
 FLAGS_SER =
 FLAGS_OMP = -openmp
diff --git a/thirdparty/swan/macros.inc.stampede2 b/thirdparty/swan/macros.inc.stampede2
index 651ae8a..dfa2734 100644
--- a/thirdparty/swan/macros.inc.stampede2
+++ b/thirdparty/swan/macros.inc.stampede2
@@ -5,7 +5,7 @@ F90_SER = ifort
 F90_OMP = ifort
 F90_MPI = mpif90
 FLAGS_OPT = -O2
-FLAGS_MSC = -W0 -assume byterecl -traceback -diag-disable 8290 -diag-disable 8291 -diag-disable 8293 -assume buffered_io -xCORE-AVX2 -axCORE-AVX512,MIC-AVX512
+FLAGS_MSC = -W0 -assume byterecl -traceback -diag-disable 8290 -diag-disable 8291 -diag-disable 8293 -xCORE-AVX2 -axCORE-AVX512,MIC-AVX512
 FLAGS90_MSC = $(FLAGS_MSC)
 FLAGS_SER =
 FLAGS_OMP = -openmp
diff --git a/thirdparty/swan/macros.inc.supermic b/thirdparty/swan/macros.inc.supermic
index 94d190e..a95927e 100644
--- a/thirdparty/swan/macros.inc.supermic
+++ b/thirdparty/swan/macros.inc.supermic
@@ -2,7 +2,7 @@ F90_SER = ifort
 F90_OMP = ifort
 F90_MPI = mpif90
 FLAGS_OPT = -O2
-FLAGS_MSC = -W0 -assume byterecl -traceback -diag-disable 8290 -diag-disable 8291 -diag-disable 8293 -assume buffered_io -xAVX
+FLAGS_MSC = -W0 -assume byterecl -traceback -diag-disable 8290 -diag-disable 8291 -diag-disable 8293 -xAVX
 FLAGS90_MSC = $(FLAGS_MSC)
 FLAGS_SER =
 FLAGS_OMP = -openmp
diff --git a/thirdparty/swan/platform.pl b/thirdparty/swan/platform.pl
index a56dfcc..0b9d0e2 100644
--- a/thirdparty/swan/platform.pl
+++ b/thirdparty/swan/platform.pl
@@ -56,7 +56,7 @@ elsif ($os =~ /AIX/i) {
   print OUTFILE "F90_SER = xlf90_r\n";
   print OUTFILE "F90_OMP = xlf90_r\n";
   print OUTFILE "F90_MPI = mpxlf90_r\n";
-  print OUTFILE "FLAGS_OPT = -O3 -qstrict -qarch=auto -qtune=auto -qnohot -qcache=auto \\\n";
+  print OUTFILE "FLAGS_OPT = -O2 -qstrict -qarch=auto -qtune=auto -qnohot -qcache=auto \\\n";
   print OUTFILE "            -qunroll -qalign=4k -qfloat=hsflt\n";
   print OUTFILE "FLAGS_MSC = -w -qfixed -qnosave -q64\n";
   print OUTFILE "FLAGS90_MSC = -w -qfree=f90 -qnosave -q64\n";
@@ -492,7 +492,7 @@ elsif ($os =~ /Linux/i) {
     print OUTFILE "F90_SER = xlf90\n";
     print OUTFILE "F90_OMP =\n";
     print OUTFILE "F90_MPI = mpfort\n";
-    print OUTFILE "FLAGS_OPT = -O3 -qstrict -qarch=auto -qtune=auto\n";
+    print OUTFILE "FLAGS_OPT = -O2 -qstrict -qarch=auto -qtune=auto\n";
     print OUTFILE "FLAGS_MSC = -qfixed -qzerosize -qwarn64\n";
     print OUTFILE "FLAGS90_MSC = -qfree=f90 -qzerosize -qwarn64\n";
     print OUTFILE "FLAGS_DYN =\n";
@@ -729,7 +729,7 @@ elsif ($os =~ /Darwin/i) {
     print OUTFILE "F90_SER = xlf90\n";
     print OUTFILE "F90_OMP =\n";
     print OUTFILE "F90_MPI =\n";
-    print OUTFILE "FLAGS_OPT = -O3 -qstrict -qtune=auto -qcache=auto -qalign=4k\n";
+    print OUTFILE "FLAGS_OPT = -O2 -qstrict -qtune=auto -qcache=auto -qalign=4k\n";
     print OUTFILE "FLAGS_MSC = -w -qfixed\n";
     print OUTFILE "FLAGS90_MSC = -w -qfree=f90\n";
     print OUTFILE "FLAGS_DYN =\n";
diff --git a/work/cmplrflags.mk b/work/cmplrflags.mk
index 50d4603..36b7663 100644
--- a/work/cmplrflags.mk
+++ b/work/cmplrflags.mk
@@ -90,12 +90,9 @@ ifeq ($(compiler),gnu)
   ifeq ($(NETCDF),enable)
      ifeq ($(MACHINENAME),blueridge)
         # FLIBS       := $(FLIBS) -L$(HDF5HOME) -lhdf5
-        NETCDFHOME    :=/usr
         FFLAGS1       :=$(FFLAGS1) -I/usr/lib64/gfortran/modules
         FFLAGS2       :=$(FFLAGS1)
         FFLAGS3       :=$(FFLAGS1)
-        # NETCDFHOME  :=/shared/apps/RHEL-5/x86_64/NetCDF/netcdf-4.1.1-gcc4.1-ifort
-        # NETCDFHOME  :=/shared/apps/RHEL-5/x86_64/NetCDF/netcdf-4.1.2-gcc4.1-ifort
         FLIBS          :=$(FLIBS) -L/usr/lib64 -lnetcdff
      else
         FLIBS          := $(FLIBS) -L$(HDF5HOME) -lhdf5 -lhdf5_fortran
@@ -159,20 +156,17 @@ ifeq ($(compiler),gfortran)
         #
         # export LD_LIBRARY_PATH=${LD_LIBRARY_PATH}:${HOME}/local/lib
         # export PATH=${PATH}:${HOME}/local/bin
-        NETCDFHOME := ${HOME}/local
         FFLAGS1 := $(FFLAGS1) -I${HOME}/local/include
         FFLAGS2 := $(FFLAGS2) -I${HOME}/local/include
         FFLAGS3 := $(FFLAGS3) -I${HOME}/local/include
         FLIBS := $(FLIBS) -L${HOME}/local/lib -lnetcdff -lnetcdf
      endif
      ifeq ($(MACHINENAME),jason-desktop)
-        NETCDFHOME := /usr
         FFLAGS1 := $(FFLAGS1) -L/usr/lib/x86_64-linux-gnu
         FFLAGS2 := $(FFLAGS2) -L/usr/lib/x86_64-linux-gnu
         FFLAGS3 := $(FFLAGS3) -L/usr/lib/x86_64-linux-gnu
      endif
      ifeq ($(MACHINENAME),rostam)
-        NETCDFHOME := /usr
         FFLAGS1 := $(FFLAGS1) -I/usr/lib64/gfortran/modules
         FFLAGS2 := $(FFLAGS2) -I/usr/lib64/gfortran/modules
         FFLAGS3 := $(FFLAGS3) -I/usr/lib64/gfortran/modules
@@ -200,7 +194,7 @@ ifeq ($(compiler),g95)
   PPFC		:=  g95
   FC		:=  g95
   PFC		:=  mpif90
-  FFLAGS1	:=  $(INCDIRS) -O3 -mcmodel=medium -fstatic -ffixed-line-length-132
+  FFLAGS1	:=  $(INCDIRS) -O2 -mcmodel=medium -fstatic -ffixed-line-length-132
   FFLAGS2	:=  $(FFLAGS1)
   FFLAGS3	:=  $(FFLAGS1)
   DA		:=  -DREAL8 -DLINUX -DCSCA
@@ -229,7 +223,7 @@ ifeq ($(compiler),intel)
   PPFC            :=  ifort
   FC            :=  ifort
   PFC           :=  mpif90
-  FFLAGS1       :=  $(INCDIRS) -O2 -FI -assume byterecl -132 -xSSE4.2 -assume buffered_io
+  FFLAGS1       :=  $(INCDIRS) -O2 -FI -assume byterecl -132 -xSSE4.2
   CFLAGS        := $(INCDIRS) -O2 -xSSE4.2 -m64 -mcmodel=medium -DLINUX
   FLIBS         :=
   ifeq ($(DEBUG),full)
@@ -252,44 +246,88 @@ ifeq ($(compiler),intel)
   endif
   #
   ifeq ($(MACHINENAME),stampede2)
-     FFLAGS1 := $(INCDIRS) -O3 -FI -assume byterecl -132 -xCORE-AVX2 -axCORE-AVX512,MIC-AVX512 -assume buffered_io
-     CFLAGS  := $(INCDIRS) -O3 -DLINUX -xCORE-AVX2 -axCORE-AVX512,MIC-AVX512
+     FFLAGS1 := $(INCDIRS) -O2 -FI -assume byterecl -132 -xCORE-AVX2 -axCORE-AVX512,MIC-AVX512
+     CFLAGS  := $(INCDIRS) -O2 -DLINUX -xCORE-AVX2 -axCORE-AVX512,MIC-AVX512
      FLIBS   := $(INCDIRS) -xCORE-AVX2 -axCORE-AVX512,MIC-AVX512
      ifeq ($(DEBUG),trace)
-        FFLAGS1 := $(INCDIRS) -g -O0 -traceback -FI -assume byterecl -132 -xCORE-AVX2 -axCORE-AVX512,MIC-AVX512 -assume buffered_io
+        FFLAGS1 := $(INCDIRS) -g -O0 -traceback -FI -assume byterecl -132 -xCORE-AVX2 -axCORE-AVX512,MIC-AVX512
         CFLAGS  := $(INCDIRS) -g -O0 -traceback -DLINUX -xCORE-AVX2 -axCORE-AVX512,MIC-AVX512
         FLIBS   := $(INCDIRS) -xCORE-AVX2 -axCORE-AVX512,MIC-AVX512
      endif
   endif
   ifeq ($(MACHINENAME),frontera) 
-     FFLAGS1 := $(INCDIRS) -O3 -FI -assume byterecl -132 -xCORE-AVX512 -assume buffered_io
-     CFLAGS  := $(INCDIRS) -O3 -DLINUX -xCORE-AVX512 
+     FFLAGS1 := $(INCDIRS) -O2 -FI -assume byterecl -132 -xCORE-AVX512
+     CFLAGS  := $(INCDIRS) -O2 -DLINUX -xCORE-AVX512 
      FLIBS   := $(INCDIRS) -xCORE-AVX512 
      ifeq ($(DEBUG),trace)
-        FFLAGS1 := $(INCDIRS) -g -O0 -traceback -FI -assume byterecl -132 -xCORE-AVX512 -assume buffered_io
+        FFLAGS1 := $(INCDIRS) -g -O0 -traceback -FI -assume byterecl -132 -xCORE-AVX512
         CFLAGS  := $(INCDIRS) -g -O0 -traceback -DLINUX -xCORE-AVX512
         FLIBS   := $(INCDIRS) -xCORE-AVX512 
      endif
   endif
   ifeq ($(MACHINENAME),queenbee)
-     FFLAGS1 := $(INCDIRS) -O3 -FI -assume byterecl -132 -xSSE4.2 -assume buffered_io
-     CFLAGS  := $(INCDIRS) -O3 -DLINUX -xSSE4.2
+     FFLAGS1 := $(INCDIRS) -O2 -FI -assume byterecl -132 -xSSE4.2
+     CFLAGS  := $(INCDIRS) -O2 -DLINUX -xSSE4.2
      FLIBS   := $(INCDIRS) -xSSE4.2
      ifeq ($(DEBUG),trace)
-        FFLAGS1 := $(INCDIRS) -g -O0 -traceback -FI -assume byterecl -132 -xSSE4.2 -assume buffered_io
+        FFLAGS1 := $(INCDIRS) -g -O0 -traceback -FI -assume byterecl -132 -xSSE4.2
         CFLAGS  := $(INCDIRS) -g -O0 -traceback -DLINUX -xSSE4.2
         FLIBS   := $(INCDIRS) -xSSE4.2
      endif
+  endif
   ifeq ($(MACHINENAME),supermic) 
-     FFLAGS1 := $(INCDIRS) -O3 -FI -assume byterecl -132 -xAVX -assume buffered_io
-     CFLAGS  := $(INCDIRS) -O3 -DLINUX -xAVX
+     FFLAGS1 := $(INCDIRS) -O2 -FI -assume byterecl -132 -xAVX
+     CFLAGS  := $(INCDIRS) -O2 -DLINUX -xAVX
      FLIBS   := $(INCDIRS) -xAVX
      ifeq ($(DEBUG),trace)
-        FFLAGS1 := $(INCDIRS) -g -O0 -traceback -FI -assume byterecl -132 -xAVX -assume buffered_io
+        FFLAGS1 := $(INCDIRS) -g -O0 -traceback -FI -assume byterecl -132 -xAVX
         CFLAGS  := $(INCDIRS) -g -O0 -traceback -DLINUX  -xAVX 
         FLIBS   := $(INCDIRS) -xAVX
      endif
   endif
+  ifeq ($(MACHINENAME),rostam) 
+     FFLAGS1 := $(INCDIRS) -O2 -FI -assume byterecl -132 -xAVX
+     CFLAGS  := $(INCDIRS) -O2 -DLINUX -xAVX
+     FLIBS   := $(INCDIRS) -xAVX
+     ifeq ($(DEBUG),trace)
+        FFLAGS1 := $(INCDIRS) -g -O0 -traceback -FI -assume byterecl -132 -xAVX
+        CFLAGS  := $(INCDIRS) -g -O0 -traceback -DLINUX  -xAVX 
+        FLIBS   := $(INCDIRS) -xAVX
+     endif
+  endif
+  ifeq ($(MACHINENAME),queenbeeC) 
+     FFLAGS1 := $(INCDIRS) -O2 -FI -assume byterecl -132 -xCORE-AVX512
+     CFLAGS  := $(INCDIRS) -O2 -DLINUX -xCORE-AVX512 
+     FLIBS   := $(INCDIRS) -xCORE-AVX512 
+     ifeq ($(DEBUG),trace)
+        FFLAGS1 := $(INCDIRS) -g -O0 -traceback -FI -assume byterecl -132 -xCORE-AVX512
+        CFLAGS  := $(INCDIRS) -g -O0 -traceback -DLINUX -xCORE-AVX512 
+        FLIBS   := $(INCDIRS) -xCORE-AVX512 
+     endif
+  endif
+  ifeq ($(MACHINENAME),ls6)
+     FFLAGS1 := $(INCDIRS) -O2 -FI -assume byterecl -132
+     CFLAGS  := $(INCDIRS) -O2 -DLINUX
+     FLIBS   := $(INCDIRS)
+     ifeq ($(DEBUG),trace)
+        FFLAGS1 := $(INCDIRS) -g -O0 -traceback -FI -assume byterecl -132
+        CFLAGS  := $(INCDIRS) -g -O0 -traceback -DLINUX
+        FLIBS   := $(INCDIRS)
+     endif
+  endif
+  #wwlwpd: NOTE: Mike III at LSU-HPC recommends 'mpiifort', Intel's MPI implementation wrapper
+  #wwlwpd: instead of the traditional "mpif90"; but the flags are the same we still redefine
+  #wwlwpd: "PFC" below
+  ifeq ($(MACHINENAME),mike)
+     PFC     :=  mpiifort
+     FFLAGS1 := $(INCDIRS) -O2 -FI -assume byterecl -132 -xAVX
+     CFLAGS  := $(INCDIRS) -O2 -DLINUX -xAVX
+     FLIBS   := $(INCDIRS) -xAVX
+     ifeq ($(DEBUG),trace)
+        FFLAGS1 := $(INCDIRS) -g -O0 -traceback -FI -assume byterecl -132 -xAVX
+        CFLAGS  := $(INCDIRS) -g -O0 -traceback -DLINUX  -xAVX
+        FLIBS   := $(INCDIRS) -xAVX
+     endif
   endif
   #
   #@jasonfleming Added to fix bus error on hatteras@renci
@@ -311,34 +349,27 @@ ifeq ($(compiler),intel)
   MSGLIBS       :=
   ifeq ($(NETCDF),enable)
      ifeq ($(MACHINENAME),hatteras)
-        NETCDFHOME  :=$(shell nc-config --prefix)
-        FLIBS       :=$(FLIBS) -L$(NETCDFHOME)/lib -lnetcdff -lnetcdf
-        FFLAGS1     :=$(FFLAGS1) -I$(NETCDFHOME)/include
-        FFLAGS2     :=$(FFLAGS1)
-        FFLAGS3     :=$(FFLAGS1)
+        FLIBS      := $(FLIBS) -L$(NETCDFHOME)/lib -lnetcdff -lnetcdf
      endif
      # jgf20150417 queenbee requires that the analyst load the netcdf and
      # netcdf_fortran modules prior to compiling or executing ADCIRC
      ifeq ($(MACHINENAME),queenbee)
-        FLIBS       := $(FLIBS) -L/usr/local/packages/netcdf/4.2.1.1/INTEL-140-MVAPICH2-2.0/lib -lnetcdff -lnetcdf
-        NETCDFHOME    :=/usr/local/packages/netcdf/4.2.1.1/INTEL-140-MVAPICH2-2.0
+        FLIBS      := $(FLIBS) -L$(NETCDFHOME)/lib -lnetcdff -lnetcdf
+     endif
+     ifeq ($(MACHINENAME),queenbeeC)
+        FLIBS      := $(FLIBS) -L$(NETCDFHOME)/lib -lnetcdff -lnetcdf
      endif
      ifeq ($(MACHINENAME),supermic)
-        FLIBS      := $(FLIBS) -L /usr/local/packages/netcdf_fortran/4.2/INTEL-140-MVAPICH2-2.0/lib -lnetcdff -L/usr/local/packages/netcdf/4.2.1.1/INTEL-140-MVAPICH2-2.0/lib -lnetcdf -lnetcdf -liomp5 -lpthread
-        NETCDFHOME :=/usr/local/packages/netcdf/4.2.1.1/INTEL-140-MVAPICH2-2.0/include
-        FFLAGS1    :=$(FFLAGS1) -I/usr/local/packages/hdf5/1.8.12/INTEL-140-MVAPICH2-2.0/include
+        FLIBS      := $(FLIBS) -L$(NETCDFHOME)/lib -lnetcdff -lnetcdf
+     endif
+     ifeq ($(MACHINENAME),rostam)
+        FLIBS      := $(FLIBS) -L$(NETCDFHOME)/lib -lnetcdff -lnetcdf
      endif
      ifeq ($(MACHINENAME),stampede)
-        NETCDFHOME :=/opt/apps/intel17/netcdf/4.3.3.1/x86_64
         FLIBS      := $(FLIBS) -L$(NETCDFHOME)/lib -lnetcdff -lnetcdf
      endif
      ifeq ($(MACHINENAME),stampede2)
-        NETCDFHOME :=/opt/apps/intel17/netcdf/4.3.3.1/x86_64
         FLIBS      := $(FLIBS) -L$(NETCDFHOME)/lib -lnetcdff -lnetcdf
-        ifeq ($(USER),jgflemin)
-           NETCDFHOME :=/work/00976/jgflemin/stampede2/local
-           FLIBS      := $(FLIBS) -L$(NETCDFHOME)/lib -lnetcdff -lnetcdf
-        endif
      endif
      ifeq ($(MACHINENAME),frontera)
         # specify NETCDFHOME on the command line or as an environment var
@@ -347,16 +378,18 @@ ifeq ($(compiler),intel)
      # @jasonfleming: Added support for lonestar5 at tacc.utexas.edu;
      # load the following module: netcdf/4.3.3.1
      ifeq ($(MACHINENAME),lonestar5)
-        #NETCDFHOME :=/opt/apps/intel18/netcdf/4.3.3.1/x86_64
-        # @jasonfleming: Updated support for lonestar5
-        NETCDFHOME :=/opt/apps/intel18/netcdf/4.6.2/x86_64
+        FLIBS      := $(FLIBS) -L$(NETCDFHOME)/lib -lnetcdff -lnetcdf
+     endif
+     ifeq ($(MACHINENAME),ls6)
+        FLIBS      := $(FLIBS) -L$(NETCDFHOME)/lib -lnetcdff -lnetcdf
+     endif
+     ifeq ($(MACHINENAME),mike)
         FLIBS      := $(FLIBS) -L$(NETCDFHOME)/lib -lnetcdff -lnetcdf
      endif
      # jgf20150817: Adding support for spirit.afrl.hpc.mil;
      # load the following modules: netcdf-fortran/intel/4.4.2
      # and hdf5/intel/1.8.12 and hdf5-mpi/intel/sgimpt/1.8.12
      ifeq ($(MACHINENAME),spirit)
-        NETCDFHOME :=/app/wpostool/COST/netcdf-fortran-4.4.2/intel
         FLIBS      := $(FLIBS) -L/app/wpostool/COST/netcdf-c-4.3.1.1/intel/lib -L$(NETCDFHOME)/lib -L/app/wpostool/COST/hdf5-mpi/1.8.12/intel/sgimpt/lib -lnetcdff -lnetcdf
      endif
      # jgf20150420 mike requires that the analyst add netcdf to the softenv
@@ -364,16 +397,13 @@ ifeq ($(compiler),intel)
      # soft add +netcdf-4.1.3-Intel-13.0.0
      ifeq ($(MACHINENAME),mike)
         FLIBS       := $(FLIBS) -L/usr/local/packages/netcdf/4.1.3/Intel-13.0.0/lib -lnetcdff -lnetcdf
-        NETCDFHOME    :=/usr/local/packages/netcdf/4.1.3/Intel-13.0.0
      endif
      ifeq ($(MACHINENAME),killdevil)
         HDF5HOME       :=/nas02/apps/hdf5-1.8.5/lib
-        NETCDFHOME     :=/nas02/apps/netcdf-4.1.1
         FLIBS          := $(FLIBS) -L$(HDF5HOME) -lhdf5 -lhdf5_fortran
      endif
   endif
   #jgf20110519: For netcdf on topsail at UNC, use
-  #NETCDFHOME=/ifs1/apps/netcdf/
   $(warning (INFO) Corresponding machine found in cmplrflags.mk.)
   ifneq ($(FOUND),TRUE)
      FOUND := TRUE
@@ -387,7 +417,7 @@ ifeq ($(compiler),intel-ND)
   PPFC          :=  ifort
   FC            :=  ifort
   PFC           :=  mpif90
-  FFLAGS1       :=  $(INCDIRS) -w -O3 -assume byterecl -132 -assume buffered_io #-i-dynamic
+  FFLAGS1       :=  $(INCDIRS) -w -O2 -assume byterecl -132 #-i-dynamic
   ifeq ($(DEBUG),full)
      FFLAGS1    :=  $(INCDIRS) -g -O0 -traceback -debug -check all -FI -assume byterecl -132 -DEBUG -DALL_TRACE -DFULL_STACK -DFLUSH_MESSAGES
   endif
@@ -402,7 +432,7 @@ ifeq ($(compiler),intel-ND)
   IMODS         :=  -I
   CC            := icc
   CCBE          := $(CC)
-  CFLAGS        := $(INCDIRS) -O3 -m64 -mcmodel=medium -DLINUX
+  CFLAGS        := $(INCDIRS) -O2 -m64 -mcmodel=medium -DLINUX
   FLIBS         := 
   ifeq ($(DATETIME),enable)
      DATETIMEHOME  := $(SRCDIR)/lib/datetime-fortran-master/build/
@@ -428,7 +458,6 @@ ifeq ($(compiler),intel-ND)
   else
      MULTIPLE := TRUE
   endif
-  NETCDFHOME=/afs/crc.nd.edu/x86_64_linux/n/netcdf/4.7.0/intel/18.0/
 endif
 # SGI ICE X (e.g. topaz@ERDC) using Intel compilers, added by TCM
 # jgf: Added flags for Thunder@AFRL.
@@ -438,7 +467,7 @@ ifeq ($(compiler),intel-sgi)
   PFC           :=  mpif90
   CC            :=  icc -O2 -no-ipo
   CCBE          :=  icc -O2 -no-ipo
-  FFLAGS1       :=  $(INCDIRS) -fixed -extend-source 132 -O2 -finline-limit=1000 -real-size 64 -no-ipo -assume buffered_io
+  FFLAGS1       :=  $(INCDIRS) -fixed -extend-source 132 -O2 -finline-limit=1000 -real-size 64 -no-ipo
 #  FFLAGS1      :=  $(INCDIRS) -Mextend -g -O0 -traceback
   FFLAGS2       :=  $(FFLAGS1)
   FFLAGS3       :=  $(FFLAGS1) -assume buffered_stdout
@@ -449,17 +478,6 @@ ifeq ($(compiler),intel-sgi)
   IMODS         :=  -module
   FLIBS         :=
   ifeq ($(NETCDF),enable)
-     ifeq ($(MACHINENAME),topaz)
-        NETCDFHOME  :=/apps/unsupported/netcdf/4.3.3.1-intel-15.0.3
-     endif
-     ifeq ($(MACHINENAME),thunder)
-        # add the following lines to ~/.personal.bashrc:
-        # module load costinit
-        # module load git
-        # module load netcdf-fortran/intel/4.4.2
-        NETCDFHOME  :=/app/COST/netcdf-fortran/4.4.2/intel
-     endif
-     # for platforms other than topaz, specify NETCDFHOME on the command line
      FLIBS       := $(FLIBS) -lnetcdff
   endif
   MSGLIBS       :=
@@ -493,7 +511,6 @@ ifeq ($(compiler),cray_xt3)
 # When compiling with netCDF support, the HDF5 libraries must also
 # be linked in, so the user must specify HDF5HOME on the command line.
 # jgf20101102: on Sapphire,
-#              NETCDFHOME=/usr/local/usp/PETtools/CE/pkgs/netcdf-4.1.1-serial
 #              HDF5HOME=${PET_HOME}/pkgs/hdf5-1.8.5-serial/lib
   ifeq ($(NETCDF),enable)
      FLIBS          := $(FLIBS) -L$(HDF5HOME) -lhdf5_fortran -lhdf5_hl -lhdf5 -lz
@@ -536,10 +553,8 @@ ifeq ($(compiler),cray_xt4)
 # be linked in, so the user must specify HDF5HOME on the command line.
 # On Jade, HDF5 was compiled with szip compression, so this library is
 # required as well.
-# jgf20101102: on Jade, NETCDFHOME=/usr/local/usp/PETtools/CE/pkgs/netcdf-4.0.1-serial
 # jgf20101102: on Jade, HDF5HOME=${PET_HOME}/pkgs/hdf5-1.8.4-serial/lib
 # jgf20101103: on Jade, SZIPHOME=/usr/local/usp/PETtools/CE/pkgs/szip-2.1/lib
-# jgf20110728: on Garnet, NETCDFHOME=/opt/cray/netcdf/4.1.1.0/netcdf-pgi
   ifeq ($(NETCDF),enable)
      FLIBS          := $(FLIBS) -L$(HDF5HOME) -L$(SZIPHOME) -lhdf5_fortran -lhdf5_hl -lhdf5 -lsz -lz
   endif
@@ -572,7 +587,6 @@ ifeq ($(compiler),cray_xt5)
   FLIBS  	:=
 # When compiling with netCDF support, the HDF5 libraries must also
 # be linked in, so the user must specify HDF5HOME on the command line.
-# jgf20090518: on Jade, NETCDFHOME=/usr/local/usp/PETtools/CE/pkgs/netcdf-4.0
 # jgf20090518: on Jade, HDF5HOME=${PET_HOME}/pkgs/hdf5-1.8.2/lib
   ifeq ($(NETCDF),enable)
      FLIBS          := $(FLIBS) -L$(HDF5HOME) -lhdf5 -lhdf5_fortran
@@ -594,7 +608,7 @@ ifeq ($(compiler),xtintel)
   PFC           :=  ftn
   CC            :=  cc -O2 -no-ipo
   CCBE          :=  cc -O2 -no-ipo
-  FFLAGS1       :=  $(INCDIRS) -fixed -extend-source 132 -O2 -default64 -finline-limit=1000 -real-size 64 -no-ipo -assume buffered_io
+  FFLAGS1       :=  $(INCDIRS) -fixed -extend-source 132 -O2 -default64 -finline-limit=1000 -real-size 64 -no-ipo
 #  FFLAGS1      :=  $(INCDIRS) -Mextend -g -O0 -traceback
   FFLAGS2       :=  $(FFLAGS1)
   FFLAGS3       :=  $(FFLAGS1) -assume buffered_stdout
@@ -606,7 +620,6 @@ ifeq ($(compiler),xtintel)
   FLIBS         :=
 # When compiling with netCDF support, the HDF5 libraries must also
 # be linked in, so the user must specify HDF5HOME on the command line.
-# jgf20090518: on Jade, NETCDFHOME=/usr/local/usp/PETtools/CE/pkgs/netcdf-4.0
 # jgf20090518: on Jade, HDF5HOME=${PET_HOME}/pkgs/hdf5-1.8.2/lib
   ifeq ($(NETCDF),enable)
      FLIBS          := $(FLIBS) -L$(HDF5HOME) -lhdf5 -lhdf5_fortran
@@ -676,10 +689,8 @@ ifeq ($(compiler),utils)
 # be linked in, so the user must specify HDF5HOME on the command line.
 # On Jade, HDF5 was compiled with szip compression, so this library is
 # required as well.
-# jgf20101102: on Jade, NETCDFHOME=/usr/local/usp/PETtools/CE/pkgs/netcdf-4.0.1-serial
 # jgf20101102: on Jade, HDF5HOME=${PET_HOME}/pkgs/hdf5-1.8.4-serial/lib
 # jgf20101103: on Jade, SZIPHOME=/usr/local/usp/PETtools/CE/pkgs/szip-2.1/lib
-# jgf20110728: on Garnet, NETCDFHOME=/opt/cray/netcdf/4.1.1.0/netcdf-pgi
   ifeq ($(NETCDF),enable)
      FLIBS          := $(FLIBS) -L$(HDF5HOME) -L$(SZIPHOME) -lhdf5_fortran -lhdf5_hl -lhdf5 -lsz -lz
   endif
@@ -726,8 +737,8 @@ ifeq ($(compiler),diamond)
   PPFC          :=  ifort
   FC            :=  ifort
   PFC           :=  ifort
-#  FFLAGS1       :=  $(INCDIRS) -O3 -xT -132
-  FFLAGS1       := -O3 -132 -xSSSE3
+#  FFLAGS1       :=  $(INCDIRS) -O2 -xT -132
+  FFLAGS1       := -O2 -132 -xSSSE3
   ifeq ($(DEBUG),full)
      FFLAGS1	:=  $(INCDIRS) -g -O0 -debug -fpe0 -132 -traceback -check all -DALL_TRACE -DFLUSH_MESSAGES -DFULL_STACK
   endif
@@ -742,8 +753,8 @@ ifeq ($(compiler),diamond)
   IMODS         :=  -I
   CC            := icc
   CCBE          := $(CC)
-#  CFLAGS        := $(INCDIRS) -O3 -xT
-  CFLAGS        := $(INCDIRS) -O3 -xSSSE3
+#  CFLAGS        := $(INCDIRS) -O2 -xT
+  CFLAGS        := $(INCDIRS) -O2 -xSSSE3
   ifeq ($(DEBUG),full)
      CFLAGS        := $(INCDIRS) -g -O0
   endif
@@ -751,10 +762,8 @@ ifeq ($(compiler),diamond)
   LIBS          :=
 # When compiling with netCDF support, the HDF5 libraries must also
 # be linked in, so the user must specify HDF5HOME on the command line.
-# jgf20101103: on Diamond, NETCDFHOME=/usr/local/usp/PETtools/CE/pkgs/netcdf-4.0.1-serial
 # jgf20101103: on Diamond, HDF5HOME=${PET_HOME}/pkgs/hdf5-1.8.4-serial/lib
   ifeq ($(NETCDF),enable)
-     NETCDFHOME     :=/usr/local/usp/PETtools/CE/pkgs/netcdf-4.2.1.1-intel-serial
      FFLAGS1        :=$(FFLAGS1) -I/usr/local/usp/PETtools/CE/pkgs/netcdf-4.2.1.1-intel-serial/include
      FFLAGS2        :=$(FFLAGS1)
      FFLAGS3        :=$(FFLAGS1)
@@ -795,7 +804,6 @@ ifeq ($(compiler),garnet)
   endif
   IMODS		:=  -module
   FLIBS         :=
-# jgf20110728: on Garnet, NETCDFHOME=/opt/cray/netcdf/4.1.1.0/netcdf-pgi
 # jgf20110815: on Garnet, HDF5HOME=/opt/cray/hdf5/default/hdf5-pgi
 # jgf20130815: on Garnet, load module cray-netcdf, with the path to the
 #              installation being /opt/cray/netcdf/4.3.0
@@ -819,7 +827,7 @@ ifeq ($(compiler),kraken)
   PPFC          :=  ftn
   FC            :=  ftn
   PFC           :=  ftn
-  FFLAGS1       :=  $(INCDIRS) -O3 -static  -132
+  FFLAGS1       :=  $(INCDIRS) -O2 -static  -132
   FFLAGS2       :=  $(FFLAGS1)
   FFLAGS3       :=  $(FFLAGS1)
   DA            :=  -DREAL8 -DLINUX -DCSCA -DPOWELL
@@ -984,7 +992,6 @@ ifeq ($(compiler),gnu)
   FLIBS		:=
   ifeq ($(NETCDF),enable)
      ifeq ($(MACHINENAME),jason-desktop)
-        NETCDFHOME := /usr/local
         HDF5HOME   := /usr/local/hdf5/hdf5-1.8.8/hdf5
      endif
      FLIBS          := $(FLIBS) -L$(HDF5HOME)/lib -L$(NETCDFHOME) -lnetcdf -lhdf5_hl -lhdf5 -lhdf5_fortran -lz
@@ -1039,9 +1046,6 @@ ifeq ($(compiler),gfortran)
   endif
   FLIBS         :=
   ifeq ($(NETCDF),enable)
-     ifeq ($(MACHINENAME),jason-desktop)
-        NETCDFHOME := /usr
-     endif
      FLIBS      := $(FLIBS) -lnetcdff
   endif
   IMODS 	:=  -I
@@ -1105,7 +1109,7 @@ ifeq ($(arch),altix)
   PPFC            := ifort
   FC              := ifort
   PFC             := ifort
-  FFLAGS1	  := $(INCDIRS) -O3 -tpp2
+  FFLAGS1	  := $(INCDIRS) -O2 -tpp2
   FFLAGS2	  := $(FFLAGS1)
   FFLAGS3	  := $(FFLAGS1)
   DA	          :=  -DREAL8 -DCSCA
@@ -1178,7 +1182,7 @@ ifeq ($(IBM),p5)
   FFLAGS0       := $(INCDIRS) -w -qfixed=132 -qarch=auto -qcache=auto
   FFLAGS1       := $(FFLAGS0) -O2
   FFLAGS2       := $(FFLAGS0) -qhot -qstrict
-  FFLAGS3       := $(FFLAGS0) -O3 -qinitauto
+  FFLAGS3       := $(FFLAGS0) -O2 -qinitauto
   DA            := -WF,"-DREAL8,-DIBM,-DCSCA"
   DP            := -tF -WF,"-DREAL8,-DIBM,-DCSCA,-DCMPI"
   DPRE          := -tF -WF,"-DREAL8,-DIBM"
@@ -1221,7 +1225,6 @@ ifeq ($(IBM),p6)
    MSGLIBS       := -lm
 # When compiling with netCDF support, the HDF5 libraries must also
 # be linked in, so the user must specify HDF5HOME on the command line.
-# yf20110301: on Cirrus/Stratus, NETCDFHOME=/usrx/local/bin/
 # yf20110301: on Cirrus/Stratus, HDF5HOME=/usrx/local/hdf5/lib
   HDF5HOME=/usrx/local/hdf5/lib
   ifeq ($(NETCDF),enable)
@@ -1498,9 +1501,9 @@ ifneq (,$(findstring powerpc-darwin,$(MACHINE)-$(OS)))
   PPFC	        := f90
   FC	        := f90
   PFC	        := mpif77
-  FFLAGS1	:=  $(INCDIRS) -w -O3 -m64 -cpu:g5 -f fixed -W132 -I . -DLINUX
-  FFLAGS2	:=  $(INCDIRS) -w -O3 -m64 -cpu:g5 -N11 -f fixed -W132 -I .
-  FFLAGS3	:=  $(INCDIRS) -w -O3 -m64 -cpu:g5 -N11 -f fixed -W132 -I .
+  FFLAGS1	:=  $(INCDIRS) -w -O2 -m64 -cpu:g5 -f fixed -W132 -I . -DLINUX
+  FFLAGS2	:=  $(INCDIRS) -w -O2 -m64 -cpu:g5 -N11 -f fixed -W132 -I .
+  FFLAGS3	:=  $(INCDIRS) -w -O2 -m64 -cpu:g5 -N11 -f fixed -W132 -I .
   DA  	   	:=  -DREAL8 -DCSCA -DLINUX
   DP  	   	:=  -DREAL8 -DCSCA -DCMPI -DLINUX
   DPRE	   	:=  -DREAL8 -DLINUX
@@ -1528,17 +1531,17 @@ ifneq (,$(findstring i386-darwin,$(MACHINE)-$(OS)))
   PPFC	        := ifort
   FC	        := ifort
   PFC	        := mpif77
-  FFLAGS1       :=  $(INCDIRS) -nowarn -O3    -fixed -132 -check all -traceback -DLINUX -DNETCDF_DEBUG -I .
-# FFLAGS1	:=  $(INCDIRS) -nowarn -O3    -fixed -132 -DIBM -I .
-  FFLAGS2	:=  $(INCDIRS) -nowarn -O3    -fixed -132 -I .
-  FFLAGS3	:=  $(INCDIRS) -nowarn -O3    -fixed -132 -I .
+  FFLAGS1       :=  $(INCDIRS) -nowarn -O2    -fixed -132 -check all -traceback -DLINUX -DNETCDF_DEBUG -I .
+# FFLAGS1	:=  $(INCDIRS) -nowarn -O2    -fixed -132 -DIBM -I .
+  FFLAGS2	:=  $(INCDIRS) -nowarn -O2    -fixed -132 -I .
+  FFLAGS3	:=  $(INCDIRS) -nowarn -O2    -fixed -132 -I .
   DA  	   	:=  -DREAL8 -DCSCA -DLINUX
   DP  	   	:=  -DREAL8 -DCSCA -DLINUX -DCMPI -DNETCDF_DEBUG
   DPRE	   	:=  -DREAL8 -DLINUX
   IMODS  	:=  -I
   CC            :=  gcc
   CCBE          :=  $(CC)
-  CFLAGS        :=  $(INCDIRS) -O3 -DLINUX
+  CFLAGS        :=  $(INCDIRS) -O2 -DLINUX
   LDFLAGS	:=
   FLIBS	        :=
   MSGLIBS	:=
