diff --git a/work/cmplrflags.mk b/work/cmplrflags.mk
index 7863444..7976779 100644
--- a/work/cmplrflags.mk
+++ b/work/cmplrflags.mk
@@ -356,6 +356,9 @@ ifeq ($(compiler),intel)
      ifeq ($(MACHINENAME),lonestar5)
         FLIBS      := $(FLIBS) -L$(NETCDFHOME)/lib -lnetcdff -lnetcdf
      endif
+     ifeq ($(MACHINENAME),ls6)
+        FLIBS      := $(FLIBS) -L$(NETCDFHOME)/lib -lnetcdff -lnetcdf
+     endif
      # jgf20150817: Adding support for spirit.afrl.hpc.mil;
      # load the following modules: netcdf-fortran/intel/4.4.2
      # and hdf5/intel/1.8.12 and hdf5-mpi/intel/sgimpt/1.8.12
