diff --git a/thirdparty/swan/macros.inc b/thirdparty/swan/macros.inc
index bbdf24a..0958e02 100644
--- a/thirdparty/swan/macros.inc
+++ b/thirdparty/swan/macros.inc
@@ -1,42 +1,24 @@
 ##############################################################################
-# IA32_Intel/x86-64_Intel:	Intel Pentium with Linux using Intel compiler 17.
+# IA32_GNU:		Intel Pentium with Linux using GNU compiler gfortran.
 ##############################################################################
-F90_SER = ifort
-F90_OMP = ifort
-# if appropriate, use mpiifort of Intel instead
+F90_SER = gfortran
+F90_OMP = gfortran
 F90_MPI = mpif90
-FLAGS_OPT = -O2
-FLAGS_MSC = -W0 -assume byterecl -traceback -diag-disable 8290 -diag-disable 8291 -diag-disable 8293
-FLAGS90_MSC = $(FLAGS_MSC)
-FLAGS_DYN = -fPIC
+FLAGS_OPT = -O
+FLAGS_MSC = -w -fno-second-underscore -ffree-line-length-none
+FLAGS90_MSC = $(FLAGS_MSC) -ffree-line-length-none
 FLAGS_SER =
-FLAGS_OMP = -qopenmp
+FLAGS_OMP = -fopenmp
 FLAGS_MPI =
-NETCDFROOT =
-ifneq ($(NETCDFROOT),)
-  INCS_SER = -I$(NETCDFROOT)/include
-  INCS_OMP = -I$(NETCDFROOT)/include
-  INCS_MPI = -I$(NETCDFROOT)/include
-  LIBS_SER = -L$(NETCDFROOT)/lib -lnetcdf -lnetcdff
-  LIBS_OMP = -L$(NETCDFROOT)/lib -lnetcdf -lnetcdff
-  LIBS_MPI = -L$(NETCDFROOT)/lib -lnetcdf -lnetcdff
-  NCF_OBJS = nctablemd.o agioncmd.o swn_outnc.o
-else
-  INCS_SER =
-  INCS_OMP =
-  INCS_MPI =
-  LIBS_SER =
-  LIBS_OMP =
-  LIBS_MPI =
-  NCF_OBJS =
-endif
+INCS_SER =
+INCS_OMP =
+INCS_MPI =
+LIBS_SER =
+LIBS_OMP = -static-libgcc
+LIBS_MPI =
 O_DIR = ../work/odir4/
 OUT = -o 
 EXTO = o
 MAKE = make
 RM = rm -f
-ifneq ($(NETCDFROOT),)
-  swch = -unix -impi -netcdf
-else
-  swch = -unix -impi
-endif
+swch = -unix
diff --git a/thirdparty/swan/macros.inc.intel b/thirdparty/swan/macros.inc.intel
new file mode 100644
index 0000000..bbdf24a
--- /dev/null
+++ b/thirdparty/swan/macros.inc.intel
@@ -0,0 +1,42 @@
+##############################################################################
+# IA32_Intel/x86-64_Intel:	Intel Pentium with Linux using Intel compiler 17.
+##############################################################################
+F90_SER = ifort
+F90_OMP = ifort
+# if appropriate, use mpiifort of Intel instead
+F90_MPI = mpif90
+FLAGS_OPT = -O2
+FLAGS_MSC = -W0 -assume byterecl -traceback -diag-disable 8290 -diag-disable 8291 -diag-disable 8293
+FLAGS90_MSC = $(FLAGS_MSC)
+FLAGS_DYN = -fPIC
+FLAGS_SER =
+FLAGS_OMP = -qopenmp
+FLAGS_MPI =
+NETCDFROOT =
+ifneq ($(NETCDFROOT),)
+  INCS_SER = -I$(NETCDFROOT)/include
+  INCS_OMP = -I$(NETCDFROOT)/include
+  INCS_MPI = -I$(NETCDFROOT)/include
+  LIBS_SER = -L$(NETCDFROOT)/lib -lnetcdf -lnetcdff
+  LIBS_OMP = -L$(NETCDFROOT)/lib -lnetcdf -lnetcdff
+  LIBS_MPI = -L$(NETCDFROOT)/lib -lnetcdf -lnetcdff
+  NCF_OBJS = nctablemd.o agioncmd.o swn_outnc.o
+else
+  INCS_SER =
+  INCS_OMP =
+  INCS_MPI =
+  LIBS_SER =
+  LIBS_OMP =
+  LIBS_MPI =
+  NCF_OBJS =
+endif
+O_DIR = ../work/odir4/
+OUT = -o 
+EXTO = o
+MAKE = make
+RM = rm -f
+ifneq ($(NETCDFROOT),)
+  swch = -unix -impi -netcdf
+else
+  swch = -unix -impi
+endif
