--- adcirc-cg-v53release-gfortran-desktop/work/cmplrflags.mk	2022-07-20 12:04:20.813959160 -0400
+++ adcirc-cg-v53release-gfortran-9-desktop/work/cmplrflags.mk	2022-07-20 12:12:23.146735838 -0400
@@ -162,11 +162,8 @@
         FFLAGS3 := $(FFLAGS3) -I${HOME}/local/include
         FLIBS := $(FLIBS) -L${HOME}/local/lib -lnetcdff -lnetcdf
      endif 
-     ifeq ($(MACHINENAME),jason-desktop)
-        NETCDFHOME := /usr
-        FFLAGS1 := $(FFLAGS1) -L/usr/lib/x86_64-linux-gnu
-        FFLAGS2 := $(FFLAGS2) -L/usr/lib/x86_64-linux-gnu
-        FFLAGS3 := $(FFLAGS3) -L/usr/lib/x86_64-linux-gnu
+     ifeq ($(MACHINENAME),desktop)
+	FLIBS      := $(FLIBS) -L$(NETCDFHOME)/lib -lnetcdff -lnetcdf
      endif
      ifeq ($(MACHINENAME),rostam)    
         NETCDFHOME := /usr
@@ -329,7 +326,6 @@
         FLIBS   := $(INCDIRS) -xAVX
      endif
   endif
-  endif
   #
   #@jasonfleming Added to fix bus error on hatteras@renci
   ifeq ($(HEAP_ARRAYS),fix)
@@ -892,7 +888,7 @@
      MULTIPLE := TRUE
   endif
 endif
-
+endif
 #$(MACHINE)
 ########################################################################
 # Compiler flags for Linux operating system on 32bit x86 CPU
