diff --git a/swan/macros.inc.queenbeeC b/swan/macros.inc.queenbeeC
new file mode 100644
index 0000000..6def14e
--- /dev/null
+++ b/swan/macros.inc.queenbeeC
@@ -0,0 +1,37 @@
+F90_SER = ifort
+F90_OMP = ifort
+F90_MPI = mpif90
+FLAGS_OPT = -O2
+FLAGS_MSC = -W0 -assume byterecl -traceback -diag-disable 8290 -diag-disable 8291 -diag-disable 8293 -assume buffered_io -xCORE-AVX512
+FLAGS90_MSC = $(FLAGS_MSC)
+FLAGS_SER =
+FLAGS_OMP = -openmp
+FLAGS_MPI =
+NETCDFROOT =
+ifneq ($(NETCDFROOT),)
+  INCS_SER = -I$(NETCDFROOT)/include
+  INCS_OMP = -I$(NETCDFROOT)/include
+  INCS_MPI = -I$(NETCDFROOT)/include
+  LIBS_SER = -L$(NETCDFROOT)/lib -lnetcdf -lnetcdff
+  LIBS_OMP = -L$(NETCDFROOT)/lib -lnetcdf -lnetcdff
+  LIBS_MPI = -L$(NETCDFROOT)/lib -lnetcdf -lnetcdff
+  NCF_OBJS = nctablemd.o agioncmd.o swn_outnc.o
+else
+  INCS_SER =
+  INCS_OMP =
+  INCS_MPI =
+  LIBS_SER =
+  LIBS_OMP =
+  LIBS_MPI =
+  NCF_OBJS =
+endif
+O_DIR = ../work/odir4/
+OUT = -o 
+EXTO = o
+MAKE = make
+RM = rm -f
+ifneq ($(NETCDFROOT),)
+  swch = -unix -impi -netcdf
+else
+  swch = -unix -impi
+endif
diff --git a/work/cmplrflags.mk b/work/cmplrflags.mk
index 2c0eb06..d95c766 100644
--- a/work/cmplrflags.mk
+++ b/work/cmplrflags.mk
@@ -277,6 +277,16 @@ ifeq ($(compiler),intel)
         CFLAGS  := $(INCDIRS) -g -O0 -traceback -DLINUX -xSSE4.2 
         FLIBS   := $(INCDIRS) -xSSE4.2 
      endif
+  endif
+  ifeq ($(MACHINENAME),queenbeeC) 
+     FFLAGS1 := $(INCDIRS) -O3 -FI -assume byterecl -132 -xCORE-AVX512 -assume buffered_io
+     CFLAGS  := $(INCDIRS) -O3 -DLINUX -xCORE-AVX512 
+     FLIBS   := $(INCDIRS) -xCORE-AVX512 
+     ifeq ($(DEBUG),trace)
+        FFLAGS1 := $(INCDIRS) -g -O0 -traceback -FI -assume byterecl -132 -xCORE-AVX512 -assume buffered_io
+        CFLAGS  := $(INCDIRS) -g -O0 -traceback -DLINUX -xCORE-AVX512 
+        FLIBS   := $(INCDIRS) -xCORE-AVX512 
+     endif
   ifeq ($(MACHINENAME),supermic) 
      FFLAGS1 := $(INCDIRS) -O3 -FI -assume byterecl -132 -xAVX -assume buffered_io
      CFLAGS  := $(INCDIRS) -O3 -DLINUX -xAVX
@@ -318,7 +328,11 @@ ifeq ($(compiler),intel)
      # netcdf_fortran modules prior to compiling or executing ADCIRC
      ifeq ($(MACHINENAME),queenbee)
         FLIBS       := $(FLIBS) -L/usr/local/packages/netcdf/4.2.1.1/INTEL-140-MVAPICH2-2.0/lib -lnetcdff -lnetcdf
-        NETCDFHOME    :=/usr/local/packages/netcdf/4.2.1.1/INTEL-140-MVAPICH2-2.0
+        NETCDFHOME  :=/usr/local/packages/netcdf/4.2.1.1/INTEL-140-MVAPICH2-2.0
+     endif
+     ifeq ($(MACHINENAME),queenbeeC)
+        FLIBS       := $(FLIBS) -L/usr/local/packages/netcdf/4.2.1.1/INTEL-140-MVAPICH2-2.0/lib -lnetcdff -lnetcdf
+        NETCDFHOME  :=/usr/local/packages/netcdf/4.2.1.1/INTEL-140-MVAPICH2-2.0
      endif
      ifeq ($(MACHINENAME),supermic)
         FLIBS      := $(FLIBS) -L /usr/local/packages/netcdf_fortran/4.2/INTEL-140-MVAPICH2-2.0/lib -lnetcdff -L/usr/local/packages/netcdf/4.2.1.1/INTEL-140-MVAPICH2-2.0/lib -lnetcdf -lnetcdf -liomp5 -lpthread
