diff --git a/swan/platform.pl b/swan/platform.pl
index 92842ed..dc15158 100644
--- a/swan/platform.pl
+++ b/swan/platform.pl
@@ -262,6 +262,52 @@ elsif ($os =~ /Linux/i) {
     print OUTFILE "  swch = -unix -impi\n";
     print OUTFILE "endif\n";
   }
+  elsif ( $compiler eq "ifx" )
+  {
+    print OUTFILE <<EOF;
+    ##############################################################################
+    # x86-64_Intel:	Intel Pentium with Linux using Intel OneAPI Compiler    .
+    ##############################################################################
+    F90_SER = ifx
+    F90_OMP = ifx
+    F90_MPI = mpiifx
+    FLAGS_OPT = -O2
+    FLAGS_MSC = -W0 -assume byterecl -traceback -diag-disable 8290 -diag-disable 8291 -diag-disable 8293
+    FLAGS90_MSC = \$(FLAGS_MSC)
+    FLAGS_DYN = -fPIC
+    FLAGS_SER =
+    FLAGS_OMP = -openmp
+    FLAGS_MPI =
+    NETCDFROOT =
+    ifneq (\$(NETCDFROOT),)
+      INCS_SER = -I\$(NETCDFROOT)/include
+      INCS_OMP = -I\$(NETCDFROOT)/include
+      INCS_MPI = -I\$(NETCDFROOT)/include
+      LIBS_SER = -L\$(NETCDFROOT)/lib -lnetcdf -lnetcdff
+      LIBS_OMP = -L\$(NETCDFROOT)/lib -lnetcdf -lnetcdff
+      LIBS_MPI = -L\$(NETCDFROOT)/lib -lnetcdf -lnetcdff
+      NCF_OBJS = nctablemd.o agioncmd.o swn_outnc.o
+    else
+      INCS_SER =
+      INCS_OMP =
+      INCS_MPI =
+      LIBS_SER =
+      LIBS_OMP =
+      LIBS_MPI =
+      NCF_OBJS =
+    endif
+    O_DIR = ../work/odir4/
+    OUT = -o
+    EXTO = o
+    MAKE = make
+    RM = rm -f
+    ifneq (\$(NETCDFROOT),)
+      swch = -unix -impi -netcdf
+    else
+      swch = -unix -impi
+    endif
+EOF
+  }
   elsif ( $compiler eq "ifc" )
   {
     print OUTFILE "##############################################################################\n";
@@ -686,7 +732,7 @@ sub getcmpl {
    my $compiler = $ENV{'FC'};
 
    unless ( $compiler ) {
-      foreach ('ifort','f90','ifc','efc','pgf90','xlf90', 'lf95','gfortran','g95') {
+      foreach ('ifx','ifort','f90','ifc','efc','pgf90','xlf90', 'lf95','gfortran','g95') {
          $compiler = $_;
          my $path  = `which $compiler`;
          last if $path;
