diff --git a/work/makefile b/work/makefile
index b823fda..e0236b5 100644
--- a/work/makefile
+++ b/work/makefile
@@ -160,7 +160,10 @@ ifeq ($(BUILDTYPE),adcswan)
     endif
   endif
   $(shell echo "SWANLASTCOMPILE := serial" > $(SRCDIR)/work/.swanlastcompile)
+  $(warning (INFO) perl ../swan/switch.pl $(swch) -adcirc ../swan/*.ftn ../swan/*.ftn90)
   $(shell perl ../swan/switch.pl $(swch) -adcirc ../swan/*.ftn ../swan/*.ftn90)
+  $(warning (INFO) perl ../swan/platform.pl && mv -f ./macros.inc ../swan/macros.inc)
+  $(shell perl ../swan/platform.pl && mv -f ./macros.inc ../swan/macros.inc)
 endif
 #                                      $(PARALLEL_ADCIRC)
 ifeq ($(BUILDTYPE),$(PARALLEL_ADCIRC))
@@ -199,7 +202,10 @@ ifeq ($(BUILDTYPE),padcswan)
     endif
   endif
   $(shell echo "SWANLASTCOMPILE := parallel" > $(SRCDIR)/work/.swanlastcompile)
+  $(warning (INFO) perl ../swan/switch.pl $(swch) -pun -adcirc ../swan/*.ftn ../swan/*.ftn90)
   $(shell perl ../swan/switch.pl $(swch) -pun -adcirc ../swan/*.ftn ../swan/*.ftn90)
+  $(warning (INFO) perl ../swan/platform.pl && mv -f ./macros.inc ../swan/macros.inc)
+  $(shell perl ../swan/platform.pl && mv -f ./macros.inc ../swan/macros.inc)
 endif
 #                                    libadc.a
 ifeq ($(BUILDTYPE),$(LIBADC))
@@ -482,7 +488,6 @@ ifeq ($(MAKELEVEL),0)
    adcswan:
 	$(MAKE) BUILDTYPE=adcswan CC="$(CCBE)" CFLAGS="$(CFLAGS)" $@
    padcswan:
-	@perl ../swan/switch.pl $(swch) -pun -adcirc ../swan/*.ftn ../swan/*.ftn90
 	$(MAKE) BUILDTYPE=padcswan CC="$(CCBE)" CFLAGS="$(CFLAGS)" $@
    libadc: $(LIBADC)
    $(LIBADC):
