diff --git a/swan/macros.inc.kraken b/swan/macros.inc.kraken
index a8cf30b..7245699 100644
--- a/swan/macros.inc.kraken
+++ b/swan/macros.inc.kraken
@@ -7,9 +7,9 @@ F90_OMP = ftn
 F90_MPI = ftn 
 FLAGS_OPT =
 FLAGS_MSC = 
-FLAGS_SER = -O3 -static 
+FLAGS_SER = -O2 -static 
 FLAGS_OMP = -mp
-FLAGS_MPI = -O3 -static 
+FLAGS_MPI = -O2 -static 
 INCS_SER =
 INCS_OMP =
 INCS_MPI =
diff --git a/swan/platform.pl b/swan/platform.pl
index 92842ed..57161bf 100644
--- a/swan/platform.pl
+++ b/swan/platform.pl
@@ -56,7 +56,7 @@ elsif ($os =~ /AIX/i) {
   print OUTFILE "F90_SER = xlf90_r\n";
   print OUTFILE "F90_OMP = xlf90_r\n";
   print OUTFILE "F90_MPI = mpxlf90_r\n";
-  print OUTFILE "FLAGS_OPT = -O3 -qstrict -qarch=auto -qtune=auto -qnohot -qcache=auto \\\n";
+  print OUTFILE "FLAGS_OPT = -O2 -qstrict -qarch=auto -qtune=auto -qnohot -qcache=auto \\\n";
   print OUTFILE "            -qunroll -qalign=4k -qfloat=hsflt\n";
   print OUTFILE "FLAGS_MSC = -w -qfixed -qnosave -q64\n";
   print OUTFILE "FLAGS90_MSC = -w -qfree=f90 -qnosave -q64\n";
@@ -491,7 +491,7 @@ elsif ($os =~ /Linux/i) {
     print OUTFILE "F90_SER = xlf90\n";
     print OUTFILE "F90_OMP =\n";
     print OUTFILE "F90_MPI = mpfort\n";
-    print OUTFILE "FLAGS_OPT = -O3 -qstrict -qarch=auto -qtune=auto\n";
+    print OUTFILE "FLAGS_OPT = -O2 -qstrict -qarch=auto -qtune=auto\n";
     print OUTFILE "FLAGS_MSC = -qfixed -qzerosize -qwarn64\n";
     print OUTFILE "FLAGS90_MSC = -qfree=f90 -qzerosize -qwarn64\n";
     print OUTFILE "FLAGS_DYN =\n";
@@ -636,7 +636,7 @@ elsif ($os =~ /Darwin/i) {
   print OUTFILE "F90_SER = xlf90\n";
   print OUTFILE "F90_OMP =\n";
   print OUTFILE "F90_MPI =\n";
-  print OUTFILE "FLAGS_OPT = -O3 -qstrict -qtune=auto -qcache=auto -qalign=4k\n";
+  print OUTFILE "FLAGS_OPT = -O2 -qstrict -qtune=auto -qcache=auto -qalign=4k\n";
   print OUTFILE "FLAGS_MSC = -w -qfixed\n";
   print OUTFILE "FLAGS90_MSC = -w -qfree=f90\n";
   print OUTFILE "FLAGS_DYN =\n";
diff --git a/work/cmplrflags.mk b/work/cmplrflags.mk
index 57c8cc7..0ab9585 100644
--- a/work/cmplrflags.mk
+++ b/work/cmplrflags.mk
@@ -197,7 +197,7 @@ ifeq ($(compiler),g95)
   PPFC		:=  g95
   FC		:=  g95
   PFC		:=  mpif90
-  FFLAGS1	:=  $(INCDIRS) -O3 -mcmodel=medium -fstatic -ffixed-line-length-132
+  FFLAGS1	:=  $(INCDIRS) -O2 -mcmodel=medium -fstatic -ffixed-line-length-132
   FFLAGS2	:=  $(FFLAGS1)
   FFLAGS3	:=  $(FFLAGS1)
   DA		:=  -DREAL8 -DLINUX -DCSCA
@@ -374,7 +374,7 @@ ifeq ($(compiler),intel-ND)
   PPFC            :=  ifort
   FC            :=  ifort
   PFC           :=  mpif90
-  FFLAGS1       :=  $(INCDIRS) -w -O3 -assume byterecl -132 -i-dynamic -assume buffered_io
+  FFLAGS1       :=  $(INCDIRS) -w -O2 -assume byterecl -132 -i-dynamic -assume buffered_io
   ifeq ($(DEBUG),full)
      FFLAGS1       :=  $(INCDIRS) -g -O0 -traceback -debug -check all -i-dynamic -FI -assume byterecl -132 -DALL_TRACE -DFULL_STACK -DFLUSH_MESSAGES
   endif
@@ -389,7 +389,7 @@ ifeq ($(compiler),intel-ND)
   IMODS         :=  -I
   CC            := icc
   CCBE          := $(CC)
-  CFLAGS        := $(INCDIRS) -O3 -m64 -mcmodel=medium -DLINUX
+  CFLAGS        := $(INCDIRS) -O2 -m64 -mcmodel=medium -DLINUX
   FLIBS          :=
   ifeq ($(DEBUG),full)
      CFLAGS        := $(INCDIRS) -g -O0 -march=k8 -m64 -mcmodel=medium -DLINUX
@@ -706,8 +706,8 @@ ifeq ($(compiler),diamond)
   PPFC          :=  ifort
   FC            :=  ifort
   PFC           :=  ifort
-#  FFLAGS1       :=  $(INCDIRS) -O3 -xT -132
-  FFLAGS1       := -O3 -132 -xSSSE3
+#  FFLAGS1       :=  $(INCDIRS) -O2 -xT -132
+  FFLAGS1       := -O2 -132 -xSSSE3
   ifeq ($(DEBUG),full)
      FFLAGS1	:=  $(INCDIRS) -g -O0 -debug -fpe0 -132 -traceback -check all -DALL_TRACE -DFLUSH_MESSAGES -DFULL_STACK
   endif
@@ -722,8 +722,8 @@ ifeq ($(compiler),diamond)
   IMODS         :=  -I
   CC            := icc
   CCBE          := $(CC)
-#  CFLAGS        := $(INCDIRS) -O3 -xT
-  CFLAGS        := $(INCDIRS) -O3 -xSSSE3
+#  CFLAGS        := $(INCDIRS) -O2 -xT
+  CFLAGS        := $(INCDIRS) -O2 -xSSSE3
   ifeq ($(DEBUG),full)
      CFLAGS        := $(INCDIRS) -g -O0
   endif
@@ -799,7 +799,7 @@ ifeq ($(compiler),kraken)
   PPFC          :=  ftn
   FC            :=  ftn
   PFC           :=  ftn
-  FFLAGS1       :=  $(INCDIRS) -O3 -static  -132
+  FFLAGS1       :=  $(INCDIRS) -O2 -static  -132
   FFLAGS2       :=  $(FFLAGS1)
   FFLAGS3       :=  $(FFLAGS1)
   DA            :=  -DREAL8 -DLINUX -DCSCA -DPOWELL
@@ -1084,7 +1084,7 @@ ifeq ($(arch),altix)
   PPFC            := ifort
   FC              := ifort
   PFC             := ifort
-  FFLAGS1	  := $(INCDIRS) -O3 -tpp2
+  FFLAGS1	  := $(INCDIRS) -O2 -tpp2
   FFLAGS2	  := $(FFLAGS1)
   FFLAGS3	  := $(FFLAGS1)
   DA	          :=  -DREAL8 -DCSCA
@@ -1157,7 +1157,7 @@ ifeq ($(IBM),p5)
   FFLAGS0       := $(INCDIRS) -w -qfixed=132 -qarch=auto -qcache=auto
   FFLAGS1       := $(FFLAGS0) -O2
   FFLAGS2       := $(FFLAGS0) -qhot -qstrict
-  FFLAGS3       := $(FFLAGS0) -O3 -qinitauto
+  FFLAGS3       := $(FFLAGS0) -O2 -qinitauto
   DA            := -WF,"-DREAL8,-DIBM,-DCSCA"
   DP            := -tF -WF,"-DREAL8,-DIBM,-DCSCA,-DCMPI"
   DPRE          := -tF -WF,"-DREAL8,-DIBM"
@@ -1477,9 +1477,9 @@ ifneq (,$(findstring powerpc-darwin,$(MACHINE)-$(OS)))
   PPFC	        := f90
   FC	        := f90
   PFC	        := mpif77
-  FFLAGS1	:=  $(INCDIRS) -w -O3 -m64 -cpu:g5 -f fixed -W132 -I . -DLINUX
-  FFLAGS2	:=  $(INCDIRS) -w -O3 -m64 -cpu:g5 -N11 -f fixed -W132 -I .
-  FFLAGS3	:=  $(INCDIRS) -w -O3 -m64 -cpu:g5 -N11 -f fixed -W132 -I .
+  FFLAGS1	:=  $(INCDIRS) -w -O2 -m64 -cpu:g5 -f fixed -W132 -I . -DLINUX
+  FFLAGS2	:=  $(INCDIRS) -w -O2 -m64 -cpu:g5 -N11 -f fixed -W132 -I .
+  FFLAGS3	:=  $(INCDIRS) -w -O2 -m64 -cpu:g5 -N11 -f fixed -W132 -I .
   DA  	   	:=  -DREAL8 -DCSCA -DLINUX
   DP  	   	:=  -DREAL8 -DCSCA -DCMPI -DLINUX
   DPRE	   	:=  -DREAL8 -DLINUX
@@ -1507,17 +1507,17 @@ ifneq (,$(findstring i386-darwin,$(MACHINE)-$(OS)))
   PPFC	        := ifort
   FC	        := ifort
   PFC	        := mpif77
-  FFLAGS1       :=  $(INCDIRS) -nowarn -O3    -fixed -132 -check all -traceback -DLINUX -DNETCDF_DEBUG -I .
-# FFLAGS1	:=  $(INCDIRS) -nowarn -O3    -fixed -132 -DIBM -I .
-  FFLAGS2	:=  $(INCDIRS) -nowarn -O3    -fixed -132 -I .
-  FFLAGS3	:=  $(INCDIRS) -nowarn -O3    -fixed -132 -I .
+  FFLAGS1       :=  $(INCDIRS) -nowarn -O2    -fixed -132 -check all -traceback -DLINUX -DNETCDF_DEBUG -I .
+# FFLAGS1	:=  $(INCDIRS) -nowarn -O2    -fixed -132 -DIBM -I .
+  FFLAGS2	:=  $(INCDIRS) -nowarn -O2    -fixed -132 -I .
+  FFLAGS3	:=  $(INCDIRS) -nowarn -O2    -fixed -132 -I .
   DA  	   	:=  -DREAL8 -DCSCA -DLINUX
   DP  	   	:=  -DREAL8 -DCSCA -DLINUX -DCMPI -DNETCDF_DEBUG
   DPRE	   	:=  -DREAL8 -DLINUX
   IMODS  	:=  -I
   CC            :=  gcc
   CCBE          :=  $(CC)
-  CFLAGS        :=  $(INCDIRS) -O3 -DLINUX
+  CFLAGS        :=  $(INCDIRS) -O2 -DLINUX
   LDFLAGS	:=
   FLIBS	        :=
   MSGLIBS	:=
