diff --git a/work/cmplrflags.mk b/work/cmplrflags.mk
index 511117b..ab80a66 100644
--- a/work/cmplrflags.mk
+++ b/work/cmplrflags.mk
@@ -305,6 +305,16 @@ ifeq ($(compiler),intel)
         FLIBS   := $(INCDIRS) -xAVX
      endif
   endif
+  ifeq ($(MACHINENAME),ls6) 
+     FFLAGS1 := $(INCDIRS) -O2 -FI -assume byterecl -132
+     CFLAGS  := $(INCDIRS) -O2 -DLINUX
+     FLIBS   := $(INCDIRS)
+     ifeq ($(DEBUG),trace)
+        FFLAGS1 := $(INCDIRS) -g -O0 -traceback -FI -assume byterecl -132
+        CFLAGS  := $(INCDIRS) -g -O0 -traceback -DLINUX
+        FLIBS   := $(INCDIRS)
+     endif
+  endif
   endif
   #
   #@jasonfleming Added to fix bus error on hatteras@renci
