diff --git a/work/cmplrflags.mk b/work/cmplrflags.mk
index 520608a..a124232 100644
--- a/work/cmplrflags.mk
+++ b/work/cmplrflags.mk
@@ -1,6 +1,47 @@
 # SRCDIR is set in makefile or on the compile line
 INCDIRS := -I. -I$(SRCDIR)/prep

+########################################################################
+# Compiler flags for Linux operating system on 64bit x86 CPU
+#
+ifeq ($(MACHINE)-$(OS),x86_64-linux-gnu)
+#
+# ***NOTE*** User must select between various Linux setups
+#            by commenting/uncommenting the appropriate compiler
+#
+#compiler=gnu
+ifeq ($(compiler),gnu)
+  PPFC		:=  gfortran
+  FC		:=  gfortran
+  PFC		:=  mpif90
+  FFLAGS1	:=  $(INCDIRS) -O2 -mcmodel=medium -ffixed-line-length-none -march=corei7-avx -m64 -static
+  FFLAGS2	:=  $(FFLAGS1)
+  FFLAGS3	:=  $(FFLAGS1)
+  DA		:=  -DREAL8 -DLINUX -DCSCA
+  DP		:=  -DREAL8 -DLINUX -DCSCA -DCMPI -DHAVE_MPI_MOD
+  DPRE		:=  -DREAL8 -DLINUX
+  IMODS 	:=  -I
+  CC		:= gcc
+  CCBE		:= $(CC)
+  CFLAGS	:= $(INCDIRS) -O2 -mcmodel=medium -DLINUX -march=corei7-avx -m64 -static-libgcc
+  CLIBS	:=
+  LIBS		:=
+  MSGLIBS	:=
+  ifeq ($(NETCDF),enable)
+        FLIBS          := $(FLIBS) -L$(HDF5HOME) -lhdf5 -lhdf5_fortran
+  endif
+  $(warning (INFO) Corresponding compilers and flags found in cmplrflags.mk.)
+  ifneq ($(FOUND),TRUE)
+     FOUND := TRUE
+  else
+     MULTIPLE := TRUE
+  endif
+endif
+endif
+#$(MACHINE)
+
+
+
 ########################################################################
 # Compiler flags for Linux operating system on 64bit x86 CPU
 #
@@ -12,7 +53,7 @@ ifeq ($(MACHINE)-$(OS),x86_64-linux-gnu)
 #compiler=gnu
 #compiler=g95
 #compiler=gfortran
-#compiler=intel
+compiler=intel
 #compiler=intel-ND
 #compiler=intel-lonestar
 #compiler=intel-sgi
@@ -33,16 +74,16 @@ ifeq ($(compiler),gnu)
   PPFC		:=  gfortran
   FC		:=  gfortran
   PFC		:=  mpif90
-  FFLAGS1	:=  $(INCDIRS) -O2 -mcmodel=medium -ffixed-line-length-none -m64
+  FFLAGS1	:=  $(INCDIRS) -O2 -mcmodel=medium -ffixed-line-length-none -march=k8 -m64
   FFLAGS2	:=  $(FFLAGS1)
   FFLAGS3	:=  $(FFLAGS1)
   DA		:=  -DREAL8 -DLINUX -DCSCA
-  DP		:=  -DREAL8 -DLINUX -DCSCA -DCMPI
+  DP		:=  -DREAL8 -DLINUX -DCSCA -DCMPI -DHAVE_MPI_MOD
   DPRE		:=  -DREAL8 -DLINUX
   IMODS 	:=  -I
   CC		:= gcc
   CCBE		:= $(CC)
-  CFLAGS	:= $(INCDIRS) -O2 -mcmodel=medium -DLINUX -m64
+  CFLAGS	:= $(INCDIRS) -O2 -mcmodel=medium -DLINUX -march=k8 -m64
   CLIBS	:=
   LIBS		:=
   MSGLIBS	:=
@@ -197,7 +238,7 @@ ifeq ($(compiler),intel)
   CFLAGS        := $(INCDIRS) -O2 -xSSE4.2 -m64 -mcmodel=medium -DLINUX
   FLIBS         :=
   ifeq ($(DEBUG),full)
-     CFLAGS        := $(INCDIRS) -g -O0 -m64 -mcmodel=medium -DLINUX
+     CFLAGS        := $(INCDIRS) -g -O0 -march=k8 -m64 -mcmodel=medium -DLINUX
   endif
   ifeq ($(DEBUG),full)
      FFLAGS1       :=  $(INCDIRS) -g -O0 -traceback -debug all -check all -ftrapuv -fpe0 -FI -assume byterecl -132 -DALL_TRACE -DFULL_STACK -DFLUSH_MESSAGES
@@ -235,6 +276,31 @@ ifeq ($(compiler),intel)
         FLIBS   := $(INCDIRS) -xCORE-AVX512
      endif
   endif
+  ifeq ($(MACHINENAME),rostam)
+     PFC     :=  mpiifort
+     FFLAGS1 := $(INCDIRS) -O2 -FI -assume byterecl -132 -xAVX
+     CFLAGS  := $(INCDIRS) -O2 -DLINUX -xAVX
+     FLIBS   := $(INCDIRS) -xAVX
+     ifeq ($(DEBUG),trace)
+        FFLAGS1 := $(INCDIRS) -g -O0 -traceback -FI -assume byterecl -132 -xAVX
+        CFLAGS  := $(INCDIRS) -g -O0 -traceback -DLINUX  -xAVX
+        FLIBS   := $(INCDIRS) -xAVX
+     endif
+  endif
+  #wwlwpd: NOTE: Mike III at LSU-HPC recommends 'mpiifort', Intel's MPI implementation wrapper
+  #wwlwpd: instead of the traditional "mpif90"; but the flags are the same we still redefine
+  #wwlwpd: "PFC" below
+  ifeq ($(MACHINENAME),mike)
+     PFC     :=  mpiifort
+     FFLAGS1 := $(INCDIRS) -O2 -FI -assume byterecl -132 -xAVX
+     CFLAGS  := $(INCDIRS) -O2 -DLINUX -xAVX
+     FLIBS   := $(INCDIRS) -xAVX
+     ifeq ($(DEBUG),trace)
+        FFLAGS1 := $(INCDIRS) -g -O0 -traceback -FI -assume byterecl -132 -xAVX
+        CFLAGS  := $(INCDIRS) -g -O0 -traceback -DLINUX  -xAVX
+        FLIBS   := $(INCDIRS) -xAVX
+     endif
+  endif
   ifeq ($(MACHINENAME),queenbee)
      FFLAGS1 := $(INCDIRS) -O3 -FI -assume byterecl -132 -xSSE4.2 -assume buffered_io
      CFLAGS  := $(INCDIRS) -O3 -DLINUX -xSSE4.2
@@ -316,6 +382,9 @@ ifeq ($(compiler),intel)
         NETCDFHOME :=/opt/apps/intel18/netcdf/4.6.2/x86_64
         FLIBS      := $(FLIBS) -L$(NETCDFHOME)/lib -lnetcdff -lnetcdf
      endif
+     ifeq ($(MACHINENAME),mike)
+        FLIBS      := $(FLIBS) -L$(NETCDFHOME)/lib -lnetcdff -lnetcdf
+     endif
      # jgf20150817: Adding support for spirit.afrl.hpc.mil;
      # load the following modules: netcdf-fortran/intel/4.4.2
      # and hdf5/intel/1.8.12 and hdf5-mpi/intel/sgimpt/1.8.12
@@ -395,7 +464,7 @@ ifeq ($(compiler),intel-ND)
      FLIBS         := $(FLIBS) -lwgrib2_api -lwgrib2 -ljasper -L$(WGRIB2HOME)
   endif
   ifeq ($(DEBUG),full)
-     CFLAGS     := $(INCDIRS) -g -O0 -m64 -mcmodel=medium -DLINUX
+     CFLAGS     := $(INCDIRS) -g -O0 -m64 -march=k8 -mcmodel=medium -DLINUX
   endif
   ifeq ($(NETCDF),enable)
      #HDF5HOME=/afs/crc.nd.edu/x86_64_linux/hdf/hdf5-1.8.6-linux-x86_64-static/lib
@@ -434,7 +503,7 @@ ifeq ($(compiler),intel-sgi)
   FFLAGS2       :=  $(FFLAGS1)
   FFLAGS3       :=  $(FFLAGS1) -assume buffered_stdout
   DA            :=  -DREAL8 -DLINUX -DCSCA
-  DP            :=  -DREAL8 -DLINUX -DCMPI -DCSCA
+  DP            :=  -DREAL8 -DLINUX -DCMPI -DHAVE_MPI_MOD -DCSCA
   DPRE          :=  -DREAL8 -DLINUX
   CFLAGS        :=  $(INCDIRS) -DLINUX
   IMODS         :=  -module
@@ -474,8 +543,8 @@ ifeq ($(compiler),cray_xt3)
   FFLAGS2	:=  $(FFLAGS1)
   FFLAGS3	:=  $(FFLAGS1) -r8 -Mr8 -Mr8intrinsics
   DA  	        :=  -DREAL8 -DLINUX -DCSCA
-  DP  	        :=  -DREAL8 -DLINUX -DCMPI -DCSCA -DDEBUG_WARN_ELEV
-#  DP  	        :=  -DREAL8 -DLINUX -DCMPI -DCSCA
+  DP  	        :=  -DREAL8 -DLINUX -DCMPI -DHAVE_MPI_MOD -DCSCA -DDEBUG_WARN_ELEV
+#  DP  	        :=  -DREAL8 -DLINUX -DCMPI -DHAVE_MPI_MOD -DCSCA
   DPRE	        :=  -DREAL8 -DLINUX
   CFLAGS	:=  -c89 $(INCDIRS) -DLINUX
   IMODS		:=  -module
@@ -512,7 +581,7 @@ ifeq ($(compiler),cray_xt4)
   FFLAGS2	:=  $(FFLAGS1)
   FFLAGS3	:=  $(FFLAGS1) -r8 -Mr8 -Mr8intrinsics
   DA  	        :=  -DREAL8 -DLINUX -DCSCA
-  DP  	        :=  -DREAL8 -DLINUX -DCMPI -DCSCA
+  DP  	        :=  -DREAL8 -DLINUX -DCMPI -DHAVE_MPI_MOD -DCSCA
   DPRE	        :=  -DREAL8 -DLINUX
   ifeq ($(SWAN),enable)
      DPRE	        :=  -DREAL8 -DLINUX -DADCSWAN
@@ -556,7 +625,7 @@ ifeq ($(compiler),cray_xt5)
   FFLAGS2	:=  $(FFLAGS1)
   FFLAGS3	:=  $(FFLAGS1) -r8 -Mr8 -Mr8intrinsics
   DA  	        :=  -DREAL8 -DLINUX -DCSCA
-  DP  	        :=  -DREAL8 -DLINUX -DCMPI -DCSCA
+  DP  	        :=  -DREAL8 -DLINUX -DCMPI -DHAVE_MPI_MOD -DCSCA
   DPRE	        :=  -DREAL8 -DLINUX
   CFLAGS	:=  -c89 $(INCDIRS) -DLINUX
   IMODS		:=  -module
@@ -590,7 +659,7 @@ ifeq ($(compiler),xtintel)
   FFLAGS2       :=  $(FFLAGS1)
   FFLAGS3       :=  $(FFLAGS1) -assume buffered_stdout
   DA            :=  -DREAL8 -DLINUX -DCSCA
-  DP            :=  -DREAL8 -DLINUX -DCMPI -DCSCA
+  DP            :=  -DREAL8 -DLINUX -DCMPI -DHAVE_MPI_MOD -DCSCA
   DPRE          :=  -DREAL8 -DLINUX
   CFLAGS        :=  $(INCDIRS) -DLINUX
   IMODS         :=  -module
@@ -652,7 +721,7 @@ ifeq ($(compiler),utils)
   FFLAGS2       :=  $(FFLAGS1)
   FFLAGS3       :=  $(FFLAGS1) -r8 -Mr8 -Mr8intrinsics
   DA            :=  -DREAL8 -DLINUX -DCSCA
-  DP            :=  -DREAL8 -DLINUX -DCMPI -DCSCA
+  DP            :=  -DREAL8 -DLINUX -DCMPI -DHAVE_MPI_MOD -DCSCA
   DPRE          :=  -DREAL8 -DLINUX
   ifeq ($(SWAN),enable)
      DPRE               :=  -DREAL8 -DLINUX -DADCSWAN
@@ -725,7 +794,7 @@ ifeq ($(compiler),diamond)
   FFLAGS2       :=  $(FFLAGS1)
   FFLAGS3       :=  $(FFLAGS1)
   DA            :=  -DREAL8 -DLINUX -DCSCA
-  DP            :=  -DREAL8 -DLINUX -DCSCA -DCMPI
+  DP            :=  -DREAL8 -DLINUX -DCSCA -DCMPI -DHAVE_MPI_MOD
   DPRE          :=  -DREAL8 -DLINUX
   ifeq ($(SWAN),enable)
      DPRE          :=  -DREAL8 -DLINUX -DADCSWAN
@@ -775,7 +844,7 @@ ifeq ($(compiler),garnet)
   FFLAGS2	:=  $(FFLAGS1)
   FFLAGS3	:=  $(FFLAGS1) #-r8 -Mr8 -Mr8intrinsics
   DA  	        :=  -DREAL8 -DLINUX -DCSCA
-  DP  	        :=  -DREAL8 -DLINUX -DCMPI -DCSCA
+  DP  	        :=  -DREAL8 -DLINUX -DCMPI -DHAVE_MPI_MOD -DCSCA
   DPRE	        :=  -DREAL8 -DLINUX
   ifeq ($(SWAN),enable)
      DPRE	        :=  -DREAL8 -DLINUX -DADCSWAN
@@ -841,7 +910,7 @@ ifeq ($(compiler),circleci)
   FFLAGS2	:=  $(FFLAGS1)
   FFLAGS3	:=  $(FFLAGS1)
   DA		:=  -DREAL8 -DLINUX -DCSCA
-  DP		:=  -DREAL8 -DLINUX -DCSCA -DCMPI
+  DP		:=  -DREAL8 -DLINUX -DCSCA -DCMPI -DHAVE_MPI_MOD
   DPRE		:=  -DREAL8 -DLINUX -DADCSWAN
   IMODS 	:=  -I
   CC		:= icx
@@ -917,7 +986,7 @@ ifeq ($(compiler),intel)
   FFLAGS2	:=  $(FFLAGS1)
   FFLAGS3	:=  $(FFLAGS1)
   DA  	        :=  -DREAL8 -DLINUX -DCSCA
-  DP  	        :=  -DREAL8 -DLINUX -DCSCA -DCMPI
+  DP  	        :=  -DREAL8 -DLINUX -DCSCA -DCMPI -DHAVE_MPI_MOD
   DPRE	        :=  -DREAL8 -DLINUX
   IMODS 	:=  -I
   CC            := icc
