diff --git a/work/cmplrflags.mk b/work/cmplrflags.mk
index 28bf127..19e2d08 100644
--- a/work/cmplrflags.mk
+++ b/work/cmplrflags.mk
@@ -331,38 +331,24 @@ ifeq ($(compiler),intel)
   MSGLIBS       :=
   ifeq ($(NETCDF),enable)
      ifeq ($(MACHINENAME),hatteras)
-        NETCDFHOME  :=$(shell nc-config --prefix)
-        FLIBS       :=$(FLIBS) -L$(NETCDFHOME)/lib -lnetcdff -lnetcdf
-        FFLAGS1     :=$(FFLAGS1) -I$(NETCDFHOME)/include
-        FFLAGS2     :=$(FFLAGS1)
-        FFLAGS3     :=$(FFLAGS1)
+        FLIBS      := $(FLIBS) -L$(NETCDFHOME)/lib -lnetcdff -L$(NETCDFHOME)/lib -lnetcdf -lnetcdf
      endif
      # jgf20150417 queenbee requires that the analyst load the netcdf and
      # netcdf_fortran modules prior to compiling or executing ADCIRC
      ifeq ($(MACHINENAME),queenbee)
-        FLIBS       := $(FLIBS) -L/usr/local/packages/netcdf/4.2.1.1/INTEL-140-MVAPICH2-2.0/lib -lnetcdff -lnetcdf
-        NETCDFHOME  :=/usr/local/packages/netcdf/4.2.1.1/INTEL-140-MVAPICH2-2.0
+        FLIBS      := $(FLIBS) -L$(NETCDFHOME)/lib -lnetcdff -L$(NETCDFHOME)/lib -lnetcdf -lnetcdf
      endif
      ifeq ($(MACHINENAME),queenbeeC)
-        FLIBS       := $(FLIBS) -L/usr/local/packages/netcdf/4.2.1.1/INTEL-140-MVAPICH2-2.0/lib -lnetcdff -lnetcdf
-        NETCDFHOME  :=/usr/local/packages/netcdf/4.2.1.1/INTEL-140-MVAPICH2-2.0
+        FLIBS      := $(FLIBS) -L$(NETCDFHOME)/lib -lnetcdff -L$(NETCDFHOME)/lib -lnetcdf -lnetcdf
      endif
      ifeq ($(MACHINENAME),supermic)
-        FLIBS      := $(FLIBS) -L /usr/local/packages/netcdf_fortran/4.2/INTEL-140-MVAPICH2-2.0/lib -lnetcdff -L/usr/local/packages/netcdf/4.2.1.1/INTEL-140-MVAPICH2-2.0/lib -lnetcdf -lnetcdf -liomp5 -lpthread
-        NETCDFHOME :=/usr/local/packages/netcdf/4.2.1.1/INTEL-140-MVAPICH2-2.0/include
-        FFLAGS1    :=$(FFLAGS1) -I/usr/local/packages/hdf5/1.8.12/INTEL-140-MVAPICH2-2.0/include
+        FLIBS      := $(FLIBS) -L$(NETCDFHOME)/lib -lnetcdff -L$(NETCDFHOME)/lib -lnetcdf -lnetcdf
      endif
      ifeq ($(MACHINENAME),stampede)
-        NETCDFHOME :=/opt/apps/intel17/netcdf/4.3.3.1/x86_64
         FLIBS      := $(FLIBS) -L$(NETCDFHOME)/lib -lnetcdff -lnetcdf
      endif
      ifeq ($(MACHINENAME),stampede2)
-        NETCDFHOME :=/opt/apps/intel17/netcdf/4.3.3.1/x86_64
         FLIBS      := $(FLIBS) -L$(NETCDFHOME)/lib -lnetcdff -lnetcdf
-        ifeq ($(USER),jgflemin)
-           NETCDFHOME :=/work/00976/jgflemin/stampede2/local
-           FLIBS      := $(FLIBS) -L$(NETCDFHOME)/lib -lnetcdff -lnetcdf
-        endif
      endif
      ifeq ($(MACHINENAME),frontera)
         # specify NETCDFHOME on the command line or as an environment var
@@ -371,9 +357,6 @@ ifeq ($(compiler),intel)
      # @jasonfleming: Added support for lonestar5 at tacc.utexas.edu;
      # load the following module: netcdf/4.3.3.1
      ifeq ($(MACHINENAME),lonestar5)
-        #NETCDFHOME :=/opt/apps/intel18/netcdf/4.3.3.1/x86_64
-        # @jasonfleming: Updated support for lonestar5
-        NETCDFHOME :=/opt/apps/intel18/netcdf/4.6.2/x86_64
         FLIBS      := $(FLIBS) -L$(NETCDFHOME)/lib -lnetcdff -lnetcdf
      endif
      # jgf20150817: Adding support for spirit.afrl.hpc.mil;
