diff --git a/work/cmplrflags.mk b/work/cmplrflags.mk
index abc13d3..95649c8 100644
--- a/work/cmplrflags.mk
+++ b/work/cmplrflags.mk
@@ -166,11 +166,6 @@ ifeq ($(compiler),gfortran)
         FFLAGS2 := $(FFLAGS2) -L/usr/lib/x86_64-linux-gnu
         FFLAGS3 := $(FFLAGS3) -L/usr/lib/x86_64-linux-gnu
      endif
-     ifeq ($(MACHINENAME),rostam)
-        FFLAGS1 := $(FFLAGS1) -I/usr/lib64/gfortran/modules
-        FFLAGS2 := $(FFLAGS2) -I/usr/lib64/gfortran/modules
-        FFLAGS3 := $(FFLAGS3) -I/usr/lib64/gfortran/modules
-     endif
      ifeq ($(MACHINENAME),wsl2-ubuntu)
         NETCDFHOME := /usr
         FFLAGS1 := $(FFLAGS1) -I/usr/include
