diff --git a/work/cmplrflags.mk b/work/cmplrflags.mk
index 73ebe14..63d6147 100644
--- a/work/cmplrflags.mk
+++ b/work/cmplrflags.mk
@@ -354,6 +354,9 @@ ifeq ($(compiler),intel)
      ifeq ($(MACHINENAME),stampede2)
         FLIBS      := $(FLIBS) -L$(NETCDFHOME)/lib -lnetcdff -lnetcdf
      endif
+     ifeq ($(MACHINENAME),mike)
+        FLIBS      := $(FLIBS) -L$(NETCDFHOME)/lib -lnetcdff -lnetcdf
+     endif
      ifeq ($(MACHINENAME),frontera)
         # specify NETCDFHOME on the command line or as an environment var
         FLIBS      := $(FLIBS) -L$(NETCDFHOME)/lib -lnetcdff -lnetcdf
@@ -372,12 +375,6 @@ ifeq ($(compiler),intel)
      ifeq ($(MACHINENAME),spirit)
         FLIBS      := $(FLIBS) -L/app/wpostool/COST/netcdf-c-4.3.1.1/intel/lib -L$(NETCDFHOME)/lib -L/app/wpostool/COST/hdf5-mpi/1.8.12/intel/sgimpt/lib -lnetcdff -lnetcdf
      endif
-     # jgf20150420 mike requires that the analyst add netcdf to the softenv
-     # with the following on the command line
-     # soft add +netcdf-4.1.3-Intel-13.0.0
-     ifeq ($(MACHINENAME),mike)
-        FLIBS       := $(FLIBS) -L/usr/local/packages/netcdf/4.1.3/Intel-13.0.0/lib -lnetcdff -lnetcdf
-     endif
      ifeq ($(MACHINENAME),killdevil)
         HDF5HOME       :=/nas02/apps/hdf5-1.8.5/lib
         FLIBS          := $(FLIBS) -L$(HDF5HOME) -lhdf5 -lhdf5_fortran
