diff --git a/work/cmplrflags.mk b/work/cmplrflags.mk
index 50d4603..4ebfa10 100644
--- a/work/cmplrflags.mk
+++ b/work/cmplrflags.mk
@@ -90,12 +90,9 @@ ifeq ($(compiler),gnu)
   ifeq ($(NETCDF),enable)
      ifeq ($(MACHINENAME),blueridge)
         # FLIBS       := $(FLIBS) -L$(HDF5HOME) -lhdf5
-        NETCDFHOME    :=/usr
         FFLAGS1       :=$(FFLAGS1) -I/usr/lib64/gfortran/modules
         FFLAGS2       :=$(FFLAGS1)
         FFLAGS3       :=$(FFLAGS1)
-        # NETCDFHOME  :=/shared/apps/RHEL-5/x86_64/NetCDF/netcdf-4.1.1-gcc4.1-ifort
-        # NETCDFHOME  :=/shared/apps/RHEL-5/x86_64/NetCDF/netcdf-4.1.2-gcc4.1-ifort
         FLIBS          :=$(FLIBS) -L/usr/lib64 -lnetcdff
      else
         FLIBS          := $(FLIBS) -L$(HDF5HOME) -lhdf5 -lhdf5_fortran
@@ -159,24 +156,16 @@ ifeq ($(compiler),gfortran)
         #
         # export LD_LIBRARY_PATH=${LD_LIBRARY_PATH}:${HOME}/local/lib
         # export PATH=${PATH}:${HOME}/local/bin
-        NETCDFHOME := ${HOME}/local
         FFLAGS1 := $(FFLAGS1) -I${HOME}/local/include
         FFLAGS2 := $(FFLAGS2) -I${HOME}/local/include
         FFLAGS3 := $(FFLAGS3) -I${HOME}/local/include
         FLIBS := $(FLIBS) -L${HOME}/local/lib -lnetcdff -lnetcdf
      endif
      ifeq ($(MACHINENAME),jason-desktop)
-        NETCDFHOME := /usr
         FFLAGS1 := $(FFLAGS1) -L/usr/lib/x86_64-linux-gnu
         FFLAGS2 := $(FFLAGS2) -L/usr/lib/x86_64-linux-gnu
         FFLAGS3 := $(FFLAGS3) -L/usr/lib/x86_64-linux-gnu
      endif
-     ifeq ($(MACHINENAME),rostam)
-        NETCDFHOME := /usr
-        FFLAGS1 := $(FFLAGS1) -I/usr/lib64/gfortran/modules
-        FFLAGS2 := $(FFLAGS2) -I/usr/lib64/gfortran/modules
-        FFLAGS3 := $(FFLAGS3) -I/usr/lib64/gfortran/modules
-     endif
      FLIBS      := $(FLIBS) -lnetcdff
   endif
   IMODS 	:=  -I
@@ -280,7 +269,8 @@ ifeq ($(compiler),intel)
         CFLAGS  := $(INCDIRS) -g -O0 -traceback -DLINUX -xSSE4.2
         FLIBS   := $(INCDIRS) -xSSE4.2
      endif
-  ifeq ($(MACHINENAME),supermic) 
+  endif
+  ifeq ($(MACHINENAME),supermic)
      FFLAGS1 := $(INCDIRS) -O3 -FI -assume byterecl -132 -xAVX -assume buffered_io
      CFLAGS  := $(INCDIRS) -O3 -DLINUX -xAVX
      FLIBS   := $(INCDIRS) -xAVX
@@ -290,6 +280,49 @@ ifeq ($(compiler),intel)
         FLIBS   := $(INCDIRS) -xAVX
      endif
   endif
+  ifeq ($(MACHINENAME),rostam)
+     FFLAGS1 := $(INCDIRS) -O3 -FI -assume byterecl -132 -xAVX -assume buffered_io
+     CFLAGS  := $(INCDIRS) -O3 -DLINUX -xAVX
+     FLIBS   := $(INCDIRS) -xAVX
+     ifeq ($(DEBUG),trace)
+        FFLAGS1 := $(INCDIRS) -g -O0 -traceback -FI -assume byterecl -132 -xAVX -assume buffered_io
+        CFLAGS  := $(INCDIRS) -g -O0 -traceback -DLINUX  -xAVX 
+        FLIBS   := $(INCDIRS) -xAVX
+     endif
+  endif
+  ifeq ($(MACHINENAME),queenbeeC)
+     FFLAGS1 := $(INCDIRS) -O3 -FI -assume byterecl -132 -xCORE-AVX512 -assume buffered_io
+     CFLAGS  := $(INCDIRS) -O3 -DLINUX -xCORE-AVX512
+     FLIBS   := $(INCDIRS) -xCORE-AVX512
+     ifeq ($(DEBUG),trace)
+        FFLAGS1 := $(INCDIRS) -g -O0 -traceback -FI -assume byterecl -132 -xCORE-AVX512 -assume buffered_io
+        CFLAGS  := $(INCDIRS) -g -O0 -traceback -DLINUX -xCORE-AVX512
+        FLIBS   := $(INCDIRS) -xCORE-AVX512
+     endif
+  endif
+  ifeq ($(MACHINENAME),ls6)
+     FFLAGS1 := $(INCDIRS) -O2 -FI -assume byterecl -132
+     CFLAGS  := $(INCDIRS) -O2 -DLINUX
+     FLIBS   := $(INCDIRS)
+     ifeq ($(DEBUG),trace)
+        FFLAGS1 := $(INCDIRS) -g -O0 -traceback -FI -assume byterecl -132
+        CFLAGS  := $(INCDIRS) -g -O0 -traceback -DLINUX
+        FLIBS   := $(INCDIRS)
+     endif
+  endif
+  #wwlwpd: NOTE: Mike III at LSU-HPC recommends 'mpiifort', Intel's MPI implementation wrapper
+  #wwlwpd: instead of the traditional "mpif90"; but the flags are the same we still redefine
+  #wwlwpd: "PFC" below
+  ifeq ($(MACHINENAME),mike)
+     PFC     :=  mpiifort
+     FFLAGS1 := $(INCDIRS) -O2 -FI -assume byterecl -132 -xAVX
+     CFLAGS  := $(INCDIRS) -O2 -DLINUX -xAVX
+     FLIBS   := $(INCDIRS) -xAVX
+     ifeq ($(DEBUG),trace)
+        FFLAGS1 := $(INCDIRS) -g -O0 -traceback -FI -assume byterecl -132 -xAVX
+        CFLAGS  := $(INCDIRS) -g -O0 -traceback -DLINUX  -xAVX
+        FLIBS   := $(INCDIRS) -xAVX
+     endif
   endif
   #
   #@jasonfleming Added to fix bus error on hatteras@renci
@@ -311,34 +344,27 @@ ifeq ($(compiler),intel)
   MSGLIBS       :=
   ifeq ($(NETCDF),enable)
      ifeq ($(MACHINENAME),hatteras)
-        NETCDFHOME  :=$(shell nc-config --prefix)
-        FLIBS       :=$(FLIBS) -L$(NETCDFHOME)/lib -lnetcdff -lnetcdf
-        FFLAGS1     :=$(FFLAGS1) -I$(NETCDFHOME)/include
-        FFLAGS2     :=$(FFLAGS1)
-        FFLAGS3     :=$(FFLAGS1)
+        FLIBS      := $(FLIBS) -L$(NETCDFHOME)/lib -lnetcdff -lnetcdf
      endif
      # jgf20150417 queenbee requires that the analyst load the netcdf and
      # netcdf_fortran modules prior to compiling or executing ADCIRC
      ifeq ($(MACHINENAME),queenbee)
-        FLIBS       := $(FLIBS) -L/usr/local/packages/netcdf/4.2.1.1/INTEL-140-MVAPICH2-2.0/lib -lnetcdff -lnetcdf
-        NETCDFHOME    :=/usr/local/packages/netcdf/4.2.1.1/INTEL-140-MVAPICH2-2.0
+        FLIBS      := $(FLIBS) -L$(NETCDFHOME)/lib -lnetcdff -lnetcdf
+     endif
+     ifeq ($(MACHINENAME),queenbeeC)
+        FLIBS      := $(FLIBS) -L$(NETCDFHOME)/lib -lnetcdff -lnetcdf
      endif
      ifeq ($(MACHINENAME),supermic)
-        FLIBS      := $(FLIBS) -L /usr/local/packages/netcdf_fortran/4.2/INTEL-140-MVAPICH2-2.0/lib -lnetcdff -L/usr/local/packages/netcdf/4.2.1.1/INTEL-140-MVAPICH2-2.0/lib -lnetcdf -lnetcdf -liomp5 -lpthread
-        NETCDFHOME :=/usr/local/packages/netcdf/4.2.1.1/INTEL-140-MVAPICH2-2.0/include
-        FFLAGS1    :=$(FFLAGS1) -I/usr/local/packages/hdf5/1.8.12/INTEL-140-MVAPICH2-2.0/include
+        FLIBS      := $(FLIBS) -L$(NETCDFHOME)/lib -lnetcdff -lnetcdf
+     endif
+     ifeq ($(MACHINENAME),rostam)
+        FLIBS      := $(FLIBS) -L$(NETCDFHOME)/lib -lnetcdff -lnetcdf
      endif
      ifeq ($(MACHINENAME),stampede)
-        NETCDFHOME :=/opt/apps/intel17/netcdf/4.3.3.1/x86_64
         FLIBS      := $(FLIBS) -L$(NETCDFHOME)/lib -lnetcdff -lnetcdf
      endif
      ifeq ($(MACHINENAME),stampede2)
-        NETCDFHOME :=/opt/apps/intel17/netcdf/4.3.3.1/x86_64
         FLIBS      := $(FLIBS) -L$(NETCDFHOME)/lib -lnetcdff -lnetcdf
-        ifeq ($(USER),jgflemin)
-           NETCDFHOME :=/work/00976/jgflemin/stampede2/local
-           FLIBS      := $(FLIBS) -L$(NETCDFHOME)/lib -lnetcdff -lnetcdf
-        endif
      endif
      ifeq ($(MACHINENAME),frontera)
         # specify NETCDFHOME on the command line or as an environment var
@@ -347,16 +373,18 @@ ifeq ($(compiler),intel)
      # @jasonfleming: Added support for lonestar5 at tacc.utexas.edu;
      # load the following module: netcdf/4.3.3.1
      ifeq ($(MACHINENAME),lonestar5)
-        #NETCDFHOME :=/opt/apps/intel18/netcdf/4.3.3.1/x86_64
-        # @jasonfleming: Updated support for lonestar5
-        NETCDFHOME :=/opt/apps/intel18/netcdf/4.6.2/x86_64
+        FLIBS      := $(FLIBS) -L$(NETCDFHOME)/lib -lnetcdff -lnetcdf
+     endif
+     ifeq ($(MACHINENAME),ls6)
+        FLIBS      := $(FLIBS) -L$(NETCDFHOME)/lib -lnetcdff -lnetcdf
+     endif
+     ifeq ($(MACHINENAME),mike)
         FLIBS      := $(FLIBS) -L$(NETCDFHOME)/lib -lnetcdff -lnetcdf
      endif
      # jgf20150817: Adding support for spirit.afrl.hpc.mil;
      # load the following modules: netcdf-fortran/intel/4.4.2
      # and hdf5/intel/1.8.12 and hdf5-mpi/intel/sgimpt/1.8.12
      ifeq ($(MACHINENAME),spirit)
-        NETCDFHOME :=/app/wpostool/COST/netcdf-fortran-4.4.2/intel
         FLIBS      := $(FLIBS) -L/app/wpostool/COST/netcdf-c-4.3.1.1/intel/lib -L$(NETCDFHOME)/lib -L/app/wpostool/COST/hdf5-mpi/1.8.12/intel/sgimpt/lib -lnetcdff -lnetcdf
      endif
      # jgf20150420 mike requires that the analyst add netcdf to the softenv
@@ -364,16 +392,13 @@ ifeq ($(compiler),intel)
      # soft add +netcdf-4.1.3-Intel-13.0.0
      ifeq ($(MACHINENAME),mike)
         FLIBS       := $(FLIBS) -L/usr/local/packages/netcdf/4.1.3/Intel-13.0.0/lib -lnetcdff -lnetcdf
-        NETCDFHOME    :=/usr/local/packages/netcdf/4.1.3/Intel-13.0.0
      endif
      ifeq ($(MACHINENAME),killdevil)
         HDF5HOME       :=/nas02/apps/hdf5-1.8.5/lib
-        NETCDFHOME     :=/nas02/apps/netcdf-4.1.1
         FLIBS          := $(FLIBS) -L$(HDF5HOME) -lhdf5 -lhdf5_fortran
      endif
   endif
   #jgf20110519: For netcdf on topsail at UNC, use
-  #NETCDFHOME=/ifs1/apps/netcdf/
   $(warning (INFO) Corresponding machine found in cmplrflags.mk.)
   ifneq ($(FOUND),TRUE)
      FOUND := TRUE
@@ -428,7 +453,6 @@ ifeq ($(compiler),intel-ND)
   else
      MULTIPLE := TRUE
   endif
-  NETCDFHOME=/afs/crc.nd.edu/x86_64_linux/n/netcdf/4.7.0/intel/18.0/
 endif
 # SGI ICE X (e.g. topaz@ERDC) using Intel compilers, added by TCM
 # jgf: Added flags for Thunder@AFRL.
@@ -449,9 +473,6 @@ ifeq ($(compiler),intel-sgi)
   IMODS         :=  -module
   FLIBS         :=
   ifeq ($(NETCDF),enable)
-     ifeq ($(MACHINENAME),topaz)
-        NETCDFHOME  :=/apps/unsupported/netcdf/4.3.3.1-intel-15.0.3
-     endif
      ifeq ($(MACHINENAME),thunder)
         # add the following lines to ~/.personal.bashrc:
         # module load costinit
@@ -493,7 +514,6 @@ ifeq ($(compiler),cray_xt3)
 # When compiling with netCDF support, the HDF5 libraries must also
 # be linked in, so the user must specify HDF5HOME on the command line.
 # jgf20101102: on Sapphire,
-#              NETCDFHOME=/usr/local/usp/PETtools/CE/pkgs/netcdf-4.1.1-serial
 #              HDF5HOME=${PET_HOME}/pkgs/hdf5-1.8.5-serial/lib
   ifeq ($(NETCDF),enable)
      FLIBS          := $(FLIBS) -L$(HDF5HOME) -lhdf5_fortran -lhdf5_hl -lhdf5 -lz
@@ -536,10 +556,8 @@ ifeq ($(compiler),cray_xt4)
 # be linked in, so the user must specify HDF5HOME on the command line.
 # On Jade, HDF5 was compiled with szip compression, so this library is
 # required as well.
-# jgf20101102: on Jade, NETCDFHOME=/usr/local/usp/PETtools/CE/pkgs/netcdf-4.0.1-serial
 # jgf20101102: on Jade, HDF5HOME=${PET_HOME}/pkgs/hdf5-1.8.4-serial/lib
 # jgf20101103: on Jade, SZIPHOME=/usr/local/usp/PETtools/CE/pkgs/szip-2.1/lib
-# jgf20110728: on Garnet, NETCDFHOME=/opt/cray/netcdf/4.1.1.0/netcdf-pgi
   ifeq ($(NETCDF),enable)
      FLIBS          := $(FLIBS) -L$(HDF5HOME) -L$(SZIPHOME) -lhdf5_fortran -lhdf5_hl -lhdf5 -lsz -lz
   endif
@@ -572,7 +590,6 @@ ifeq ($(compiler),cray_xt5)
   FLIBS  	:=
 # When compiling with netCDF support, the HDF5 libraries must also
 # be linked in, so the user must specify HDF5HOME on the command line.
-# jgf20090518: on Jade, NETCDFHOME=/usr/local/usp/PETtools/CE/pkgs/netcdf-4.0
 # jgf20090518: on Jade, HDF5HOME=${PET_HOME}/pkgs/hdf5-1.8.2/lib
   ifeq ($(NETCDF),enable)
      FLIBS          := $(FLIBS) -L$(HDF5HOME) -lhdf5 -lhdf5_fortran
@@ -606,7 +623,6 @@ ifeq ($(compiler),xtintel)
   FLIBS         :=
 # When compiling with netCDF support, the HDF5 libraries must also
 # be linked in, so the user must specify HDF5HOME on the command line.
-# jgf20090518: on Jade, NETCDFHOME=/usr/local/usp/PETtools/CE/pkgs/netcdf-4.0
 # jgf20090518: on Jade, HDF5HOME=${PET_HOME}/pkgs/hdf5-1.8.2/lib
   ifeq ($(NETCDF),enable)
      FLIBS          := $(FLIBS) -L$(HDF5HOME) -lhdf5 -lhdf5_fortran
@@ -676,10 +692,8 @@ ifeq ($(compiler),utils)
 # be linked in, so the user must specify HDF5HOME on the command line.
 # On Jade, HDF5 was compiled with szip compression, so this library is
 # required as well.
-# jgf20101102: on Jade, NETCDFHOME=/usr/local/usp/PETtools/CE/pkgs/netcdf-4.0.1-serial
 # jgf20101102: on Jade, HDF5HOME=${PET_HOME}/pkgs/hdf5-1.8.4-serial/lib
 # jgf20101103: on Jade, SZIPHOME=/usr/local/usp/PETtools/CE/pkgs/szip-2.1/lib
-# jgf20110728: on Garnet, NETCDFHOME=/opt/cray/netcdf/4.1.1.0/netcdf-pgi
   ifeq ($(NETCDF),enable)
      FLIBS          := $(FLIBS) -L$(HDF5HOME) -L$(SZIPHOME) -lhdf5_fortran -lhdf5_hl -lhdf5 -lsz -lz
   endif
@@ -751,7 +765,6 @@ ifeq ($(compiler),diamond)
   LIBS          :=
 # When compiling with netCDF support, the HDF5 libraries must also
 # be linked in, so the user must specify HDF5HOME on the command line.
-# jgf20101103: on Diamond, NETCDFHOME=/usr/local/usp/PETtools/CE/pkgs/netcdf-4.0.1-serial
 # jgf20101103: on Diamond, HDF5HOME=${PET_HOME}/pkgs/hdf5-1.8.4-serial/lib
   ifeq ($(NETCDF),enable)
      NETCDFHOME     :=/usr/local/usp/PETtools/CE/pkgs/netcdf-4.2.1.1-intel-serial
@@ -795,7 +808,6 @@ ifeq ($(compiler),garnet)
   endif
   IMODS		:=  -module
   FLIBS         :=
-# jgf20110728: on Garnet, NETCDFHOME=/opt/cray/netcdf/4.1.1.0/netcdf-pgi
 # jgf20110815: on Garnet, HDF5HOME=/opt/cray/hdf5/default/hdf5-pgi
 # jgf20130815: on Garnet, load module cray-netcdf, with the path to the
 #              installation being /opt/cray/netcdf/4.3.0
@@ -1221,7 +1233,6 @@ ifeq ($(IBM),p6)
    MSGLIBS       := -lm
 # When compiling with netCDF support, the HDF5 libraries must also
 # be linked in, so the user must specify HDF5HOME on the command line.
-# yf20110301: on Cirrus/Stratus, NETCDFHOME=/usrx/local/bin/
 # yf20110301: on Cirrus/Stratus, HDF5HOME=/usrx/local/hdf5/lib
   HDF5HOME=/usrx/local/hdf5/lib
   ifeq ($(NETCDF),enable)
