--- a/perlbrew/bin/patchperl
+++ b/perlbrew/bin/patchperl
@@ -534,13 +534,22 @@ $fatpacked{"Devel/PatchPerl.pm"} = '#line '.(1+__LINE__).' "'.__FILE__."\"\n".<<
   sub _patch
   {
     my($patch) = @_;
-    print "patching $_\n" for $patch =~ /^\+{3}\s+(\S+)/gm;
+    my @ro = ();
+    for ($patch =~ /^\+{3}\s+(\S+)/gm) {
+      print "patching $_\n";
+      if (-r $_ and not -w $_) {
+        push @ro, $_;
+        chmod 0644, $_;
+      }
+    }
     my $diff = 'tmp.diff';
     _write_or_die($diff, $patch);
     die "No patch utility found\n" unless $patch_exe;
     local $ENV{PATCH_GET} = 0; # I can't reproduce this at all, but meh.
     _run_or_die("$patch_exe -f -s -p0 <$diff");
     unlink $diff or die "unlink $diff: $!\n";
+    # put back ro to 0444
+    chmod 0444, @ro;
   }
