diff --git a/prep/prep.F b/prep/prep.F
index c2a8ced..5ec7914 100644
--- a/prep/prep.F
+++ b/prep/prep.F
@@ -791,7 +791,9 @@ CTGA 20180524: Adjusted code to better handle out-of-order attributes in the bod
                IF (Mode.eq.0) SDNumND(iproc,k) = SDNumND(iproc,k)+1
                IF (Mode.eq.1) THEN
                   SDNode = IMAP_NOD_GL(2,NodeNum)
-                  WRITE(sdu(iproc),1100) labels(NodeNum),(AttrData(j),j=1,NumCol(curAttrInd))!tgaf13mod
+                  ! jgf: write the fulldomain node number, it will be mapped
+                  ! to the subdomain node number at run time
+                  WRITE(sdu(iproc),1100) NodeNum,(AttrData(j),j=1,NumCol(curAttrInd))!tgaf13mod
                ENDIF
             ELSE
                DO m=1, ITOTPROC(NodeNum)
@@ -803,7 +805,7 @@ CTGA 20180524: Adjusted code to better handle out-of-order attributes in the bod
                         ENDIF
                         IF (Mode.eq.1) THEN
                            SDNode = IMAP_NOD_GL2(2*(m-1)+2,NodeNum)
-                           WRITE(sdu(iproc),1100) labels(NodeNum),
+                           WRITE(sdu(iproc),1100) NodeNum,
      &                          (AttrData(j),j=1,NumCol(curAttrInd))!tgaf13mod
                         ENDIF
                      ENDIF
diff --git a/swan/macros.inc.queenbee b/swan/macros.inc.queenbee
index e35a794..9fd6715 100644
--- a/swan/macros.inc.queenbee
+++ b/swan/macros.inc.queenbee
@@ -5,7 +5,7 @@ F90_SER = ifort
 F90_OMP = ifort
 F90_MPI = mpif90
 FLAGS_OPT = -O2
-FLAGS_MSC = -W0 -assume byterecl -traceback -diag-disable 8290 -diag-disable 8291 -diag-disable 8293 -assume buffered_io -xSSE4.2
+FLAGS_MSC = -W0 -traceback -diag-disable 8290 -diag-disable 8291 -diag-disable 8293 -xSSE4.2
 FLAGS90_MSC = $(FLAGS_MSC)
 FLAGS_SER =
 FLAGS_OMP = -openmp
diff --git a/swan/macros.inc.stampede2 b/swan/macros.inc.stampede2
index 651ae8a..2f96d9b 100644
--- a/swan/macros.inc.stampede2
+++ b/swan/macros.inc.stampede2
@@ -5,7 +5,7 @@ F90_SER = ifort
 F90_OMP = ifort
 F90_MPI = mpif90
 FLAGS_OPT = -O2
-FLAGS_MSC = -W0 -assume byterecl -traceback -diag-disable 8290 -diag-disable 8291 -diag-disable 8293 -assume buffered_io -xCORE-AVX2 -axCORE-AVX512,MIC-AVX512
+FLAGS_MSC = -W0 -traceback -diag-disable 8290 -diag-disable 8291 -diag-disable 8293 -xCORE-AVX2 -axCORE-AVX512,MIC-AVX512
 FLAGS90_MSC = $(FLAGS_MSC)
 FLAGS_SER =
 FLAGS_OMP = -openmp
diff --git a/swan/macros.inc.supermic b/swan/macros.inc.supermic
index 94d190e..71cc03b 100644
--- a/swan/macros.inc.supermic
+++ b/swan/macros.inc.supermic
@@ -2,7 +2,7 @@ F90_SER = ifort
 F90_OMP = ifort
 F90_MPI = mpif90
 FLAGS_OPT = -O2
-FLAGS_MSC = -W0 -assume byterecl -traceback -diag-disable 8290 -diag-disable 8291 -diag-disable 8293 -assume buffered_io -xAVX
+FLAGS_MSC = -W0 -traceback -diag-disable 8290 -diag-disable 8291 -diag-disable 8293 -xAVX
 FLAGS90_MSC = $(FLAGS_MSC)
 FLAGS_SER =
 FLAGS_OMP = -openmp
diff --git a/work/cmplrflags.mk b/work/cmplrflags.mk
index 2c0eb06..6e08692 100644
--- a/work/cmplrflags.mk
+++ b/work/cmplrflags.mk
@@ -89,13 +89,10 @@ ifeq ($(compiler),gnu)
   MSGLIBS	:=
   ifeq ($(NETCDF),enable)
      ifeq ($(MACHINENAME),blueridge)
-        # FLIBS       := $(FLIBS) -L$(HDF5HOME) -lhdf5  
-        NETCDFHOME    :=/usr
+        # FLIBS       := $(FLIBS) -L$(HDF5HOME) -lhdf5
         FFLAGS1       :=$(FFLAGS1) -I/usr/lib64/gfortran/modules
         FFLAGS2       :=$(FFLAGS1)
         FFLAGS3       :=$(FFLAGS1)
-        # NETCDFHOME  :=/shared/apps/RHEL-5/x86_64/NetCDF/netcdf-4.1.1-gcc4.1-ifort
-        # NETCDFHOME  :=/shared/apps/RHEL-5/x86_64/NetCDF/netcdf-4.1.2-gcc4.1-ifort
         FLIBS          :=$(FLIBS) -L/usr/lib64 -lnetcdff
      else
         FLIBS          := $(FLIBS) -L$(HDF5HOME) -lhdf5 -lhdf5_fortran
@@ -117,9 +114,9 @@ ifeq ($(compiler),gfortran)
   PPFC		:=  gfortran
   FC		:=  gfortran
   PFC		:=  mpif90
-  FFLAGS1	:=  $(INCDIRS) -O2 -ffixed-line-length-none 
+  FFLAGS1	:=  $(INCDIRS) -O2 -ffixed-line-length-none
   ifeq ($(PROFILE),enable)
-    FFLAGS1	:=  $(INCDIRS) -pg -O0 -fprofile-arcs -ftest-coverage -ffixed-line-length-none 
+    FFLAGS1	:=  $(INCDIRS) -pg -O0 -fprofile-arcs -ftest-coverage -ffixed-line-length-none
   endif
   ifeq ($(DEBUG),full)
     FFLAGS1	:=  $(INCDIRS) -g -O0 -ffixed-line-length-none -fbacktrace -fbounds-check -ffpe-trap=zero,invalid,overflow,denormal -DALL_TRACE -DFLUSH_MESSAGES -DFULL_STACK -DDEBUG_HOLLAND -DDEBUG_WARN_ELEV
@@ -128,7 +125,7 @@ ifeq ($(compiler),gfortran)
     FFLAGS1	:=  $(INCDIRS) -g -O0 -Wall -Wextra -ffixed-line-length-none -DALL_TRACE -DFLUSH_MESSAGES -DFULL_STACK
   endif
   ifeq ($(DEBUG),full-not-warnelev)
-    FFLAGS1	:=  $(INCDIRS) -g -O0 -ffixed-line-length-none -fbacktrace -fbounds-check -ffpe-trap=zero,invalid,overflow,denormal -DALL_TRACE -DFLUSH_MESSAGES -DFULL_STACK -DDEBUG_HOLLAND 
+    FFLAGS1	:=  $(INCDIRS) -g -O0 -ffixed-line-length-none -fbacktrace -fbounds-check -ffpe-trap=zero,invalid,overflow,denormal -DALL_TRACE -DFLUSH_MESSAGES -DFULL_STACK -DDEBUG_HOLLAND
   endif
   ifeq ($(DEBUG),full-not-fpe)
     FFLAGS1	:=  $(INCDIRS) -g -O0 -ffixed-line-length-none -fbacktrace -fbounds-check -DALL_TRACE -DFLUSH_MESSAGES -DFULL_STACK -DDEBUG_HOLLAND
@@ -147,16 +144,16 @@ ifeq ($(compiler),gfortran)
   ifeq ($(SWAN),enable)
      DPRE               :=  -DREAL8 -DLINUX -DADCSWAN
   endif
-  FLIBS         := 
+  FLIBS         :=
   ifeq ($(NETCDF),enable)
-     ifeq ($(MACHINENAME),penguin)    
+     ifeq ($(MACHINENAME),penguin)
         # module purge
-        # module load gcc/6.2.0 openmpi/2.1.2/gcc.6.2.0 
+        # module load gcc/6.2.0 openmpi/2.1.2/gcc.6.2.0
         # curl -O ftp://ftp.unidata.ucar.edu/pub/netcdf/old/netcdf-4.2.1.1.tar.gz
         # curl -O ftp://ftp.unidata.ucar.edu/pub/netcdf/old/netcdf-fortran-4.2.tar.gz
         # curl -O https://support.hdfgroup.org/ftp/HDF5/releases/hdf5-1.8/hdf5-1.8.12/src/hdf5-1.8.12.tar.gz
         # CPPFLAGS=-I/home/jgflemin/local/include LDFLAGS=-L/home/jgflemin/local/lib ./configure --prefix=/home/jgflemin/local
-        # 
+        #
         # export LD_LIBRARY_PATH=${LD_LIBRARY_PATH}:${HOME}/local/lib
         # export PATH=${PATH}:${HOME}/local/bin
         NETCDFHOME := ${HOME}/local
@@ -164,20 +161,20 @@ ifeq ($(compiler),gfortran)
         FFLAGS2 := $(FFLAGS2) -I${HOME}/local/include
         FFLAGS3 := $(FFLAGS3) -I${HOME}/local/include
         FLIBS := $(FLIBS) -L${HOME}/local/lib -lnetcdff -lnetcdf
-     endif 
+     endif
      ifeq ($(MACHINENAME),jason-desktop)
         NETCDFHOME := /usr
         FFLAGS1 := $(FFLAGS1) -L/usr/lib/x86_64-linux-gnu
         FFLAGS2 := $(FFLAGS2) -L/usr/lib/x86_64-linux-gnu
         FFLAGS3 := $(FFLAGS3) -L/usr/lib/x86_64-linux-gnu
      endif
-     ifeq ($(MACHINENAME),rostam)    
+     ifeq ($(MACHINENAME),rostam)
         NETCDFHOME := /usr
         FFLAGS1 := $(FFLAGS1) -I/usr/lib64/gfortran/modules
         FFLAGS2 := $(FFLAGS2) -I/usr/lib64/gfortran/modules
         FFLAGS3 := $(FFLAGS3) -I/usr/lib64/gfortran/modules
      endif
-     FLIBS      := $(FLIBS) -lnetcdff 
+     FLIBS      := $(FLIBS) -lnetcdff
   endif
   IMODS 	:=  -I
   CC		:= gcc
@@ -200,7 +197,7 @@ ifeq ($(compiler),g95)
   PPFC		:=  g95
   FC		:=  g95
   PFC		:=  mpif90
-  FFLAGS1	:=  $(INCDIRS) -O3 -mcmodel=medium -fstatic -ffixed-line-length-132
+  FFLAGS1	:=  $(INCDIRS) -O2 -mcmodel=medium -fstatic -ffixed-line-length-132
   FFLAGS2	:=  $(FFLAGS1)
   FFLAGS3	:=  $(FFLAGS1)
   DA		:=  -DREAL8 -DLINUX -DCSCA
@@ -222,14 +219,16 @@ ifeq ($(compiler),g95)
 endif
 #
 # jgf45.12 These flags work on the UNC Topsail Cluster.
-# jgf: The -i-dynamic flag defers the inclusion of the library with 
+# jgf: The -i-dynamic flag defers the inclusion of the library with
 # feupdateenv until run time, thus avoiding the error message:
 # "feupdateenv is not implemented and will always fail"
-ifeq ($(compiler),intel)
-  PPFC            :=  ifort
+# this line is to support icc/ifort and icx/ifort (llvm based)
+ifneq (,$(filter intel intel-llvm,$(compiler)))
+  CC            :=  icc
+  PPFC          :=  ifort
   FC            :=  ifort
   PFC           :=  mpif90
-  FFLAGS1       :=  $(INCDIRS) -O2 -FI -assume byterecl -132 -xSSE4.2 -assume buffered_io
+  FFLAGS1       :=  $(INCDIRS) -O2 -FI -assume byterecl -132 -xSSE4.2
   CFLAGS        := $(INCDIRS) -O2 -xSSE4.2 -m64 -mcmodel=medium -DLINUX
   FLIBS         :=
   ifeq ($(DEBUG),full)
@@ -248,45 +247,100 @@ ifeq ($(compiler),intel)
      FFLAGS1       :=  $(INCDIRS) -g -O0 -traceback -FI -assume byterecl -132 -DNETCDF_TRACE -DFULL_STACK -DFLUSH_MESSAGES
   endif
   #
-  ifeq ($(MACHINENAME),stampede2) 
-     FFLAGS1 := $(INCDIRS) -O3 -FI -assume byterecl -132 -xCORE-AVX2 -axCORE-AVX512,MIC-AVX512 -assume buffered_io
-     CFLAGS  := $(INCDIRS) -O3 -DLINUX -xCORE-AVX2 -axCORE-AVX512,MIC-AVX512 
-     FLIBS   := $(INCDIRS) -xCORE-AVX2 -axCORE-AVX512,MIC-AVX512 
+  ifeq ($(MACHINENAME),stampede2)
+     FFLAGS1 := $(INCDIRS) -O2 -FI -assume byterecl -132 -xCORE-AVX2 -axCORE-AVX512,MIC-AVX512
+     CFLAGS  := $(INCDIRS) -O2 -DLINUX -xCORE-AVX2 -axCORE-AVX512,MIC-AVX512
+     FLIBS   := $(INCDIRS) -xCORE-AVX2 -axCORE-AVX512,MIC-AVX512
      ifeq ($(DEBUG),trace)
-        FFLAGS1 := $(INCDIRS) -g -O0 -traceback -FI -assume byterecl -132 -xCORE-AVX2 -axCORE-AVX512,MIC-AVX512 -assume buffered_io
-        CFLAGS  := $(INCDIRS) -g -O0 -traceback -DLINUX -xCORE-AVX2 -axCORE-AVX512,MIC-AVX512 
-        FLIBS   := $(INCDIRS) -xCORE-AVX2 -axCORE-AVX512,MIC-AVX512 
+        FFLAGS1 := $(INCDIRS) -g -O0 -traceback -FI -assume byterecl -132 -xCORE-AVX2 -axCORE-AVX512,MIC-AVX512
+        CFLAGS  := $(INCDIRS) -g -O0 -traceback -DLINUX -xCORE-AVX2 -axCORE-AVX512,MIC-AVX512
+        FLIBS   := $(INCDIRS) -xCORE-AVX2 -axCORE-AVX512,MIC-AVX512
      endif
   endif
-  ifeq ($(MACHINENAME),frontera) 
-     FFLAGS1 := $(INCDIRS) -O3 -FI -assume byterecl -132 -xCORE-AVX512 -assume buffered_io
-     CFLAGS  := $(INCDIRS) -O3 -DLINUX -xCORE-AVX512 
-     FLIBS   := $(INCDIRS) -xCORE-AVX512 
+  ifeq ($(MACHINENAME),frontera)
+     FFLAGS1 := $(INCDIRS) -O2 -FI -assume byterecl -132 -xCORE-AVX512
+     CFLAGS  := $(INCDIRS) -O2 -DLINUX -xCORE-AVX512
+     FLIBS   := $(INCDIRS) -xCORE-AVX512
      ifeq ($(DEBUG),trace)
-        FFLAGS1 := $(INCDIRS) -g -O0 -traceback -FI -assume byterecl -132 -xCORE-AVX512 -assume buffered_io
+        FFLAGS1 := $(INCDIRS) -g -O0 -traceback -FI -assume byterecl -132 -xCORE-AVX512
         CFLAGS  := $(INCDIRS) -g -O0 -traceback -DLINUX -xCORE-AVX512
-        FLIBS   := $(INCDIRS) -xCORE-AVX512 
+        FLIBS   := $(INCDIRS) -xCORE-AVX512
+     endif
+  endif
+  ifeq ($(MACHINENAME),queenbee)
+     FFLAGS1 := $(INCDIRS) -O2 -FI -assume byterecl -132 -xSSE4.2
+     CFLAGS  := $(INCDIRS) -O2 -DLINUX -xSSE4.2
+     FLIBS   := $(INCDIRS) -xSSE4.2
+     ifeq ($(DEBUG),trace)
+        FFLAGS1 := $(INCDIRS) -g -O0 -traceback -FI -assume byterecl -132 -xSSE4.2
+        CFLAGS  := $(INCDIRS) -g -O0 -traceback -DLINUX -xSSE4.2
+        FLIBS   := $(INCDIRS) -xSSE4.2
+     endif
+  endif
+  ifeq ($(MACHINENAME),queenbeeC)
+     FFLAGS1 := $(INCDIRS) -O2 -FI -assume byterecl -132 -xCORE-AVX512
+     CFLAGS  := $(INCDIRS) -O2 -DLINUX -xCORE-AVX512
+     FLIBS   := $(INCDIRS) -xCORE-AVX512
+     ifeq ($(DEBUG),trace)
+        FFLAGS1 := $(INCDIRS) -g -O0 -traceback -FI -assume byterecl -132 -xCORE-AVX512
+        CFLAGS  := $(INCDIRS) -g -O0 -traceback -DLINUX -xCORE-AVX512
+        FLIBS   := $(INCDIRS) -xCORE-AVX512
+     endif
+  endif
+  ifeq ($(MACHINENAME),supermic)
+     FFLAGS1 := $(INCDIRS) -O2 -FI -assume byterecl -132 -xAVX
+     CFLAGS  := $(INCDIRS) -O2 -DLINUX -xAVX
+     FLIBS   := $(INCDIRS) -xAVX
+     ifeq ($(DEBUG),trace)
+        FFLAGS1 := $(INCDIRS) -g -O0 -traceback -FI -assume byterecl -132 -xAVX
+        CFLAGS  := $(INCDIRS) -g -O0 -traceback -DLINUX  -xAVX
+        FLIBS   := $(INCDIRS) -xAVX
      endif
   endif
-  ifeq ($(MACHINENAME),queenbee) 
-     FFLAGS1 := $(INCDIRS) -O3 -FI -assume byterecl -132 -xSSE4.2 -assume buffered_io
-     CFLAGS  := $(INCDIRS) -O3 -DLINUX -xSSE4.2 
-     FLIBS   := $(INCDIRS) -xSSE4.2 
+  ifeq ($(MACHINENAME),rostam)
+     FFLAGS1 := $(INCDIRS) -O2 -FI -assume byterecl -132 -xAVX
+     CFLAGS  := $(INCDIRS) -O2 -DLINUX -xAVX
+     FLIBS   := $(INCDIRS) -xAVX
      ifeq ($(DEBUG),trace)
-        FFLAGS1 := $(INCDIRS) -g -O0 -traceback -FI -assume byterecl -132 -xSSE4.2 -assume buffered_io
-        CFLAGS  := $(INCDIRS) -g -O0 -traceback -DLINUX -xSSE4.2 
-        FLIBS   := $(INCDIRS) -xSSE4.2 
+        FFLAGS1 := $(INCDIRS) -g -O0 -traceback -FI -assume byterecl -132 -xAVX
+        CFLAGS  := $(INCDIRS) -g -O0 -traceback -DLINUX  -xAVX
+        FLIBS   := $(INCDIRS) -xAVX
      endif
-  ifeq ($(MACHINENAME),supermic) 
-     FFLAGS1 := $(INCDIRS) -O3 -FI -assume byterecl -132 -xAVX -assume buffered_io
-     CFLAGS  := $(INCDIRS) -O3 -DLINUX -xAVX
+  endif
+  ifeq ($(MACHINENAME),ls6)
+     FFLAGS1 := $(INCDIRS) -O2 -FI -assume byterecl -132
+     CFLAGS  := $(INCDIRS) -O2 -DLINUX
+     FLIBS   := $(INCDIRS)
+     ifeq ($(DEBUG),trace)
+        FFLAGS1 := $(INCDIRS) -g -O0 -traceback -FI -assume byterecl -132
+        CFLAGS  := $(INCDIRS) -g -O0 -traceback -DLINUX
+        FLIBS   := $(INCDIRS)
+     endif
+  endif
+  #wwlwpd: NOTE: Mike III at LSU-HPC recommends 'mpiifort', Intel's MPI implementation wrapper
+  #wwlwpd: instead of the traditional "mpif90"; but the flags are the same we still redefine
+  #wwlwpd: "PFC" below
+  ifeq ($(MACHINENAME),mike)
+     PFC     :=  mpiifort
+     FFLAGS1 := $(INCDIRS) -O2 -FI -assume byterecl -132 -xAVX
+     CFLAGS  := $(INCDIRS) -O2 -DLINUX -xAVX
      FLIBS   := $(INCDIRS) -xAVX
      ifeq ($(DEBUG),trace)
-        FFLAGS1 := $(INCDIRS) -g -O0 -traceback -FI -assume byterecl -132 -xAVX -assume buffered_io
-        CFLAGS  := $(INCDIRS) -g -O0 -traceback -DLINUX  -xAVX 
+        FFLAGS1 := $(INCDIRS) -g -O0 -traceback -FI -assume byterecl -132 -xAVX
+        CFLAGS  := $(INCDIRS) -g -O0 -traceback -DLINUX  -xAVX
         FLIBS   := $(INCDIRS) -xAVX
      endif
   endif
+  ifeq ($(MACHINENAME),stampede3)
+     CC      := icx
+     FFLAGS1 := $(INCDIRS) -O2 -FI -assume byterecl -132 -xCORE-AVX2 -axCORE-AVX512,COMMON-AVX512 
+     CFLAGS  := $(INCDIRS) -O2 -DLINUX -xCORE-AVX2 -axCORE-AVX512 -Wno-implicit-function-declaration
+     FLIBS   := $(INCDIRS) -xCORE-AVX2 -axCORE-AVX512,COMMON-AVX512 
+     ifeq ($(DEBUG),trace)
+        FFLAGS1 := $(INCDIRS) -g -O0 -traceback -FI -assume byterecl -132 -xCORE-AVX2 -axCORE-AVX512,COMMON-AVX512
+        CFLAGS  := $(INCDIRS) -g -O0 -traceback -DLINUX -xCORE-AVX2 -axCORE-AVX512,COMMON-AVX512 -Wimplicit-function-declaration
+        FLIBS   := $(INCDIRS) -xCORE-AVX2 -axCORE-AVX512,COMMON-AVX512
+     endif
   endif
   #
   #@jasonfleming Added to fix bus error on hatteras@renci
@@ -302,67 +356,60 @@ ifeq ($(compiler),intel)
      DPRE          := $(DPRE) -DADCSWAN
   endif
   IMODS         :=  -I
-  CC            := icc
+  #CC            := icc  # get from start of this section
   CCBE		:= $(CC)
   CLIBS         :=
   MSGLIBS       :=
   ifeq ($(NETCDF),enable)
-     ifeq ($(MACHINENAME),hatteras)
-        NETCDFHOME  :=/usr/share/Modules/software/CentOS-7/netcdf-Fortran/4.4.0_intel-18.0.0
-        FLIBS       :=$(FLIBS) -L$(NETCDFHOME)/lib -lnetcdff -lnetcdf
-        FFLAGS1     :=$(FFLAGS1) -I$(NETCDFHOME)/include
-        FFLAGS2     :=$(FFLAGS1)
-        FFLAGS3     :=$(FFLAGS1)
+     FLIBS      := $(FLIBS) -L$(NETCDFHOME)/lib -lnetcdff -lnetcdf
      endif
      # jgf20150417 queenbee requires that the analyst load the netcdf and
      # netcdf_fortran modules prior to compiling or executing ADCIRC
      ifeq ($(MACHINENAME),queenbee)
-        FLIBS       := $(FLIBS) -L/usr/local/packages/netcdf/4.2.1.1/INTEL-140-MVAPICH2-2.0/lib -lnetcdff -lnetcdf
-        NETCDFHOME    :=/usr/local/packages/netcdf/4.2.1.1/INTEL-140-MVAPICH2-2.0
+        FLIBS      := $(FLIBS) -L$(NETCDFHOME)/lib -lnetcdff -lnetcdf
+     endif
+     ifeq ($(MACHINENAME),queenbeeC)
+        FLIBS      := $(FLIBS) -L$(NETCDFHOME)/lib -lnetcdff -lnetcdf
      endif
      ifeq ($(MACHINENAME),supermic)
-        FLIBS      := $(FLIBS) -L /usr/local/packages/netcdf_fortran/4.2/INTEL-140-MVAPICH2-2.0/lib -lnetcdff -L/usr/local/packages/netcdf/4.2.1.1/INTEL-140-MVAPICH2-2.0/lib -lnetcdf -lnetcdf -liomp5 -lpthread
-        NETCDFHOME :=/usr/local/packages/netcdf/4.2.1.1/INTEL-140-MVAPICH2-2.0/include
-        FFLAGS1    :=$(FFLAGS1) -I/usr/local/packages/hdf5/1.8.12/INTEL-140-MVAPICH2-2.0/include
+        FLIBS      := $(FLIBS) -L$(NETCDFHOME)/lib -lnetcdff -lnetcdf
+     endif
+     ifeq ($(MACHINENAME),rostam)
+        FLIBS      := $(FLIBS) -L$(NETCDFHOME)/lib -lnetcdff -lnetcdf
      endif
      ifeq ($(MACHINENAME),stampede)
-        NETCDFHOME :=/opt/apps/intel17/netcdf/4.3.3.1/x86_64
         FLIBS      := $(FLIBS) -L$(NETCDFHOME)/lib -lnetcdff -lnetcdf
      endif
      ifeq ($(MACHINENAME),stampede2)
-        NETCDFHOME :=/opt/apps/intel17/netcdf/4.3.3.1/x86_64
         FLIBS      := $(FLIBS) -L$(NETCDFHOME)/lib -lnetcdff -lnetcdf
-        ifeq ($(USER),jgflemin)
-           NETCDFHOME :=/work/00976/jgflemin/stampede2/local
-           FLIBS      := $(FLIBS) -L$(NETCDFHOME)/lib -lnetcdff -lnetcdf
-        endif
+     endif
+     ifeq ($(MACHINENAME),stampede3)
+        FLIBS      := $(FLIBS) -L$(NETCDFHOME)/lib -lnetcdff -lnetcdf
      endif
      ifeq ($(MACHINENAME),frontera)
-        # specify NETCDFHOME on the command line or as an environment var
         FLIBS      := $(FLIBS) -L$(NETCDFHOME)/lib -lnetcdff -lnetcdf
      endif
      # @jasonfleming: Added support for lonestar5 at tacc.utexas.edu;
      # load the following module: netcdf/4.3.3.1
      ifeq ($(MACHINENAME),lonestar5)
-        #NETCDFHOME :=/opt/apps/intel18/netcdf/4.3.3.1/x86_64
-        # @jasonfleming: Updated support for lonestar5
-        NETCDFHOME :=/opt/apps/intel18/netcdf/4.6.2/x86_64
+        FLIBS      := $(FLIBS) -L$(NETCDFHOME)/lib -lnetcdff -lnetcdf
+     endif
+     ifeq ($(MACHINENAME),ls6)
+        FLIBS      := $(FLIBS) -L$(NETCDFHOME)/lib -lnetcdff -lnetcdf
+     endif
+     ifeq ($(MACHINENAME),mike)
         FLIBS      := $(FLIBS) -L$(NETCDFHOME)/lib -lnetcdff -lnetcdf
      endif
      # jgf20150817: Adding support for spirit.afrl.hpc.mil;
      # load the following modules: netcdf-fortran/intel/4.4.2
      # and hdf5/intel/1.8.12 and hdf5-mpi/intel/sgimpt/1.8.12
-     ifeq ($(MACHINENAME),spirit) 
+     ifeq ($(MACHINENAME),spirit)
         NETCDFHOME :=/app/wpostool/COST/netcdf-fortran-4.4.2/intel
         FLIBS      := $(FLIBS) -L/app/wpostool/COST/netcdf-c-4.3.1.1/intel/lib -L$(NETCDFHOME)/lib -L/app/wpostool/COST/hdf5-mpi/1.8.12/intel/sgimpt/lib -lnetcdff -lnetcdf
      endif
      # jgf20150420 mike requires that the analyst add netcdf to the softenv
-     # with the following on the command line 
+     # with the following on the command line
      # soft add +netcdf-4.1.3-Intel-13.0.0
-     ifeq ($(MACHINENAME),mike)
-        FLIBS       := $(FLIBS) -L/usr/local/packages/netcdf/4.1.3/Intel-13.0.0/lib -lnetcdff -lnetcdf
-        NETCDFHOME    :=/usr/local/packages/netcdf/4.1.3/Intel-13.0.0
-     endif
      ifeq ($(MACHINENAME),killdevil)
         HDF5HOME       :=/nas02/apps/hdf5-1.8.5/lib
         NETCDFHOME     :=/nas02/apps/netcdf-4.1.1
@@ -384,7 +431,7 @@ ifeq ($(compiler),intel-ND)
   PPFC            :=  ifort
   FC            :=  ifort
   PFC           :=  mpif90
-  FFLAGS1       :=  $(INCDIRS) -w -O3 -assume byterecl -132 -i-dynamic -assume buffered_io
+  FFLAGS1       :=  $(INCDIRS) -w -O2 -assume byterecl -132 -i-dynamic
   ifeq ($(DEBUG),full)
      FFLAGS1       :=  $(INCDIRS) -g -O0 -traceback -debug -check all -i-dynamic -FI -assume byterecl -132 -DALL_TRACE -DFULL_STACK -DFLUSH_MESSAGES
   endif
@@ -399,15 +446,15 @@ ifeq ($(compiler),intel-ND)
   IMODS         :=  -I
   CC            := icc
   CCBE          := $(CC)
-  CFLAGS        := $(INCDIRS) -O3 -m64 -mcmodel=medium -DLINUX
+  CFLAGS        := $(INCDIRS) -O2 -m64 -mcmodel=medium -DLINUX
   FLIBS          :=
   ifeq ($(DEBUG),full)
      CFLAGS        := $(INCDIRS) -g -O0 -march=k8 -m64 -mcmodel=medium -DLINUX
   endif
   ifeq ($(NETCDF),enable)
      HDF5HOME=/afs/crc.nd.edu/x86_64_linux/hdf/hdf5-1.8.6-linux-x86_64-static/lib
-     FLIBS      := $(FLIBS) -lnetcdff -L$(HDF5HOME) 
-  endif   
+     FLIBS      := $(FLIBS) -lnetcdff -L$(HDF5HOME)
+  endif
   CLIBS         :=
   MSGLIBS       :=
   $(warning (INFO) Corresponding machine found in cmplrflags.mk.)
@@ -428,7 +475,7 @@ ifeq ($(compiler),intel-sgi)
   PFC           :=  mpif90
   CC            :=  icc -O2 -no-ipo
   CCBE          :=  icc -O2 -no-ipo
-  FFLAGS1       :=  $(INCDIRS) -fixed -extend-source 132 -O2 -finline-limit=1000 -real-size 64 -no-ipo -assume buffered_io
+  FFLAGS1       :=  $(INCDIRS) -fixed -extend-source 132 -O2 -finline-limit=1000 -real-size 64 -no-ipo
 #  FFLAGS1      :=  $(INCDIRS) -Mextend -g -O0 -traceback
   FFLAGS2       :=  $(FFLAGS1)
   FFLAGS3       :=  $(FFLAGS1) -assume buffered_stdout
@@ -504,7 +551,7 @@ ifeq ($(compiler),cray_xt4)
   PFC	        :=  ftn
   CC		:=  pgcc
   CCBE		:=  cc
-  FFLAGS1	:=  $(INCDIRS) -Mextend -Minform,inform -O2 -fastsse 
+  FFLAGS1	:=  $(INCDIRS) -Mextend -Minform,inform -O2 -fastsse
   ifeq ($(DEBUG),full)
      FFLAGS1	:=  $(INCDIRS) -Mextend -g -O0 -traceback -Mbounds -Mchkfpstk -Mchkptr -Mchkstk -DALL_TRACE -DFLUSH_MESSAGES -DFULL_STACK
   endif
@@ -584,7 +631,7 @@ ifeq ($(compiler),xtintel)
   PFC           :=  ftn
   CC            :=  cc -O2 -no-ipo
   CCBE          :=  cc -O2 -no-ipo
-  FFLAGS1       :=  $(INCDIRS) -fixed -extend-source 132 -O2 -default64 -finline-limit=1000 -real-size 64 -no-ipo -assume buffered_io
+  FFLAGS1       :=  $(INCDIRS) -fixed -extend-source 132 -O2 -default64 -finline-limit=1000 -real-size 64 -no-ipo
 #  FFLAGS1      :=  $(INCDIRS) -Mextend -g -O0 -traceback
   FFLAGS2       :=  $(FFLAGS1)
   FFLAGS3       :=  $(FFLAGS1) -assume buffered_stdout
@@ -716,8 +763,8 @@ ifeq ($(compiler),diamond)
   PPFC          :=  ifort
   FC            :=  ifort
   PFC           :=  ifort
-#  FFLAGS1       :=  $(INCDIRS) -O3 -xT -132
-  FFLAGS1       := -O3 -132 -xSSSE3
+#  FFLAGS1       :=  $(INCDIRS) -O2 -xT -132
+  FFLAGS1       := -O2 -132 -xSSSE3
   ifeq ($(DEBUG),full)
      FFLAGS1	:=  $(INCDIRS) -g -O0 -debug -fpe0 -132 -traceback -check all -DALL_TRACE -DFLUSH_MESSAGES -DFULL_STACK
   endif
@@ -732,8 +779,8 @@ ifeq ($(compiler),diamond)
   IMODS         :=  -I
   CC            := icc
   CCBE          := $(CC)
-#  CFLAGS        := $(INCDIRS) -O3 -xT
-  CFLAGS        := $(INCDIRS) -O3 -xSSSE3
+#  CFLAGS        := $(INCDIRS) -O2 -xT
+  CFLAGS        := $(INCDIRS) -O2 -xSSSE3
   ifeq ($(DEBUG),full)
      CFLAGS        := $(INCDIRS) -g -O0
   endif
@@ -784,7 +831,7 @@ ifeq ($(compiler),garnet)
      CFLAGS	:=  $(INCDIRS) -DLINUX -g -O0
   endif
   IMODS		:=  -module
-  FLIBS         := 
+  FLIBS         :=
 # jgf20110728: on Garnet, NETCDFHOME=/opt/cray/netcdf/4.1.1.0/netcdf-pgi
 # jgf20110815: on Garnet, HDF5HOME=/opt/cray/hdf5/default/hdf5-pgi
 # jgf20130815: on Garnet, load module cray-netcdf, with the path to the
@@ -809,7 +856,7 @@ ifeq ($(compiler),kraken)
   PPFC          :=  ftn
   FC            :=  ftn
   PFC           :=  ftn
-  FFLAGS1       :=  $(INCDIRS) -O3 -static  -132
+  FFLAGS1       :=  $(INCDIRS) -O2 -static  -132
   FFLAGS2       :=  $(FFLAGS1)
   FFLAGS3       :=  $(FFLAGS1)
   DA            :=  -DREAL8 -DLINUX -DCSCA -DPOWELL
@@ -845,7 +892,7 @@ ifeq ($(compiler),circleci)
   IMODS 	:=  -I
   CC		:= gcc
   CCBE		:= $(CC)
-  CFLAGS	:= $(INCDIRS) -O0 -g -mcmodel=medium -DLINUX -m64 
+  CFLAGS	:= $(INCDIRS) -O0 -g -mcmodel=medium -DLINUX -m64
   CLIBS	:=
   LIBS		:=
   MSGLIBS	:=
@@ -859,8 +906,7 @@ ifeq ($(compiler),circleci)
      MULTIPLE := TRUE
   endif
 endif
-#
-endif
+
 #$(MACHINE)
 ########################################################################
 # Compiler flags for Linux operating system on 32bit x86 CPU
@@ -950,7 +996,7 @@ ifeq ($(compiler),gnu)
      FFLAGS1	:=  $(INCDIRS) -g -O0 -ffixed-line-length-132 -ftrace=full -fbounds-check -DNETCDF_TRACE -DFLUSH_MESSAGES -DFULL_STACK
   endif
   ifeq ($(DEBUG),valgrind)
-     FFLAGS1	:=  $(INCDIRS) -g -O0 -ffixed-line-length-132 
+     FFLAGS1	:=  $(INCDIRS) -g -O0 -ffixed-line-length-132
   endif
   ifeq ($(SWAN),enable)
      FFLAGS1    :=  $(FFLAGS1) -freal-loops
@@ -988,69 +1034,6 @@ ifeq ($(compiler),gnu)
   endif
 endif
 #
-# gfortran
-ifeq ($(compiler),gfortran)
-  ifeq ($(MACHINENAME),jason-desktop)
-     XDMFPATH    := /home/jason/projects/XDMF/Code/latestCode
-     XDMFLIBPATH := /home/jason/projects/XDMF/Code/testLatest
-  endif
-  PPFC		:=  gfortran
-  FC		:=  gfortran
-  PFC		:=  mpif90
-  FFLAGS1	:=  $(INCDIRS) -O2 -ffixed-line-length-none 
-  ifeq ($(PROFILE),enable)
-    FFLAGS1	:=  $(INCDIRS) -pg -O0 -fprofile-arcs -ftest-coverage -ffixed-line-length-none 
-  endif
-  ifeq ($(DEBUG),full)
-    FFLAGS1	:=  $(INCDIRS) -g -O0 -ffixed-line-length-none -fbacktrace -fbounds-check -ffpe-trap=zero,invalid,overflow,denormal -DALL_TRACE -DFLUSH_MESSAGES -DFULL_STACK -DDEBUG_HOLLAND -DDEBUG_WARN_ELEV
-  endif
-  ifeq ($(DEBUG),compiler-warnings)
-    FFLAGS1	:=  $(INCDIRS) -g -O0 -Wall -Wextra -ffixed-line-length-none -DALL_TRACE -DFLUSH_MESSAGES -DFULL_STACK
-  endif
-  ifeq ($(DEBUG),full-not-warnelev)
-    FFLAGS1	:=  $(INCDIRS) -g -O0 -ffixed-line-length-none -fbacktrace -fbounds-check -ffpe-trap=zero,invalid,overflow,denormal -DALL_TRACE -DFLUSH_MESSAGES -DFULL_STACK -DDEBUG_HOLLAND 
-  endif
-  ifeq ($(DEBUG),full-not-fpe)
-    FFLAGS1	:=  $(INCDIRS) -g -O0 -ffixed-line-length-none -fbacktrace -fbounds-check -DALL_TRACE -DFLUSH_MESSAGES -DFULL_STACK -DDEBUG_HOLLAND
-  endif
-  ifeq ($(DEBUG),trace)
-    FFLAGS1	:=  $(INCDIRS) -g -O0 -ffixed-line-length-none -fbacktrace -DALL_TRACE -DFLUSH_MESSAGES -DFULL_STACK
-  endif
-  ifneq ($(MACHINENAME),jason-desktop)
-     FFLAGS1 := $(FFLAGS1) -fno-underscoring
-  endif
-  FFLAGS2	:=  $(FFLAGS1)
-  FFLAGS3	:=  $(FFLAGS1)
-  DA		:=  -DREAL8 -DLINUX -DCSCA
-  DP		:=  -DREAL8 -DLINUX -DCSCA -DCMPI
-  DPRE		:=  -DREAL8 -DLINUX
-  ifeq ($(SWAN),enable)
-     DPRE               :=  -DREAL8 -DLINUX -DADCSWAN
-  endif
-  FLIBS         := 
-  ifeq ($(NETCDF),enable)
-     ifeq ($(MACHINENAME),jason-desktop)
-        NETCDFHOME := /usr
-     endif
-     FLIBS      := $(FLIBS) -lnetcdff 
-  endif
-  IMODS 	:=  -I
-  CC		:= gcc
-  CCBE		:= $(CC)
-  CFLAGS	:= $(INCDIRS) -O2 -DLINUX
-  ifeq ($(DEBUG),full)
-     CFLAGS     := $(INCDIRS) -g -O0 -DLINUX
-  endif
-  CLIBS	:=
-  MSGLIBS	:=
-  $(warning (INFO) Corresponding machine found in cmplrflags.mk.)
-  ifneq ($(FOUND),TRUE)
-     FOUND := TRUE
-  else
-     MULTIPLE := TRUE
-  endif
-endif
-
 endif
 
 ########################################################################
@@ -1095,7 +1078,7 @@ ifeq ($(arch),altix)
   PPFC            := ifort
   FC              := ifort
   PFC             := ifort
-  FFLAGS1	  := $(INCDIRS) -O3 -tpp2
+  FFLAGS1	  := $(INCDIRS) -O2 -tpp2
   FFLAGS2	  := $(FFLAGS1)
   FFLAGS3	  := $(FFLAGS1)
   DA	          :=  -DREAL8 -DCSCA
@@ -1168,7 +1151,7 @@ ifeq ($(IBM),p5)
   FFLAGS0       := $(INCDIRS) -w -qfixed=132 -qarch=auto -qcache=auto
   FFLAGS1       := $(FFLAGS0) -O2
   FFLAGS2       := $(FFLAGS0) -qhot -qstrict
-  FFLAGS3       := $(FFLAGS0) -O3 -qinitauto
+  FFLAGS3       := $(FFLAGS0) -O2 -qinitauto
   DA            := -WF,"-DREAL8,-DIBM,-DCSCA"
   DP            := -tF -WF,"-DREAL8,-DIBM,-DCSCA,-DCMPI"
   DPRE          := -tF -WF,"-DREAL8,-DIBM"
@@ -1488,9 +1471,9 @@ ifneq (,$(findstring powerpc-darwin,$(MACHINE)-$(OS)))
   PPFC	        := f90
   FC	        := f90
   PFC	        := mpif77
-  FFLAGS1	:=  $(INCDIRS) -w -O3 -m64 -cpu:g5 -f fixed -W132 -I . -DLINUX
-  FFLAGS2	:=  $(INCDIRS) -w -O3 -m64 -cpu:g5 -N11 -f fixed -W132 -I .
-  FFLAGS3	:=  $(INCDIRS) -w -O3 -m64 -cpu:g5 -N11 -f fixed -W132 -I .
+  FFLAGS1	:=  $(INCDIRS) -w -O2 -m64 -cpu:g5 -f fixed -W132 -I . -DLINUX
+  FFLAGS2	:=  $(INCDIRS) -w -O2 -m64 -cpu:g5 -N11 -f fixed -W132 -I .
+  FFLAGS3	:=  $(INCDIRS) -w -O2 -m64 -cpu:g5 -N11 -f fixed -W132 -I .
   DA  	   	:=  -DREAL8 -DCSCA -DLINUX
   DP  	   	:=  -DREAL8 -DCSCA -DCMPI -DLINUX
   DPRE	   	:=  -DREAL8 -DLINUX
@@ -1518,17 +1501,17 @@ ifneq (,$(findstring i386-darwin,$(MACHINE)-$(OS)))
   PPFC	        := ifort
   FC	        := ifort
   PFC	        := mpif77
-  FFLAGS1       :=  $(INCDIRS) -nowarn -O3    -fixed -132 -check all -traceback -DLINUX -DNETCDF_DEBUG -I .
-# FFLAGS1	:=  $(INCDIRS) -nowarn -O3    -fixed -132 -DIBM -I .
-  FFLAGS2	:=  $(INCDIRS) -nowarn -O3    -fixed -132 -I .
-  FFLAGS3	:=  $(INCDIRS) -nowarn -O3    -fixed -132 -I .
+  FFLAGS1       :=  $(INCDIRS) -nowarn -O2    -fixed -132 -check all -traceback -DLINUX -DNETCDF_DEBUG -I .
+# FFLAGS1	:=  $(INCDIRS) -nowarn -O2    -fixed -132 -DIBM -I .
+  FFLAGS2	:=  $(INCDIRS) -nowarn -O2    -fixed -132 -I .
+  FFLAGS3	:=  $(INCDIRS) -nowarn -O2    -fixed -132 -I .
   DA  	   	:=  -DREAL8 -DCSCA -DLINUX
   DP  	   	:=  -DREAL8 -DCSCA -DLINUX -DCMPI -DNETCDF_DEBUG
   DPRE	   	:=  -DREAL8 -DLINUX
   IMODS  	:=  -I
   CC            :=  gcc
   CCBE          :=  $(CC)
-  CFLAGS        :=  $(INCDIRS) -O3 -DLINUX
+  CFLAGS        :=  $(INCDIRS) -O2 -DLINUX
   LDFLAGS	:=
   FLIBS	        :=
   MSGLIBS	:=
diff --git a/work/makefile b/work/makefile
index b823fda..abb3f81 100644
--- a/work/makefile
+++ b/work/makefile
@@ -271,6 +271,17 @@ ifeq ($(BUILDTYPE),inflate)
   VPATH :=  $(SRCDIR)/util
   MSG_MOBJ:=
 endif
+#                                     adcircResultsComparison
+ifeq ($(BUILDTYPE),adcircResultsComparison)
+  CF:= $(PPFC)
+  O_DIR:=$(SRCDIR)/work/odir12/
+  FFLAGS:= $(FFLAGS2) $(IMODS) $(O_DIR) $(DA)
+  ifeq ($(NETCDF),enable)
+    LIBS  := -L$(NETCDFHOME)/lib -lnetcdf $(FLIBS)
+  endif
+  VPATH :=  $(SRCDIR)/util
+  MSG_MOBJ:=
+endif
 #                                      hstime
 ifeq ($(BUILDTYPE),hstime)
   CF:= $(PPFC)
@@ -389,6 +400,7 @@ B23_SRC    =  buildstwave23.F
 H2A_SRC    =  hot2asc.F
 INF_SRC    =  inflate.F
 HST_SRC    =  hstime.F
+RCMP_SRC   =  adcircResultsComparison.F90
 ASW_SRC    =  aswip_1.0.3.F
 CMP_SRC    =  adccmp.F
 
@@ -403,6 +415,7 @@ B12_OBJ:= $(patsubst %.F, $(O_DIR)%.o, $(B12_SRC) )
 B23_OBJ:= $(patsubst %.F, $(O_DIR)%.o, $(B23_SRC) )
 H2A_OBJ:= $(patsubst %.F, $(O_DIR)%.o, $(H2A_SRC) )
 INF_OBJ:= $(patsubst %.F, $(O_DIR)%.o, $(INF_SRC) )
+RCMP_OBJ:= $(patsubst %.F, $(O_DIR)%.o, $(RCMP_SRC) )
 HST_OBJ:= $(patsubst %.F, $(O_DIR)%.o, $(HST_SRC) )
 ASW_OBJ:= $(patsubst %.F, $(O_DIR)%.o, $(ASW_SRC) )
 CMP_OBJ:= $(patsubst %.F, $(O_DIR)%.o, $(CMP_SRC) )
@@ -455,7 +468,7 @@ $(O_DIR)%.o  : %.f90
 
 ########################## Executable Targets ############################
 
-.PHONY: all libadc clean clobber metis metis_be adcprep adcprep_be adcpost adcirc $(PARALLEL_ADCIRC)  p15 owi22 build13 build12 buildstwave23 hot2asc inflate hstime aswip adccmp adcswan padcswan
+.PHONY: all libadc clean clobber metis metis_be adcprep adcprep_be adcpost adcirc $(PARALLEL_ADCIRC)  p15 owi22 build13 build12 buildstwave23 hot2asc inflate hstime aswip adccmp adcswan padcswan adcircResultsComparison
 
 all :  metis adcprep $(BACKEND_EXEC) adcpost adcirc $(PARALLEL_ADCIRC)
 
@@ -502,6 +515,8 @@ ifeq ($(MAKELEVEL),0)
 	$(MAKE) BUILDTYPE=hot2asc $@
    inflate:
 	$(MAKE) BUILDTYPE=inflate $@
+   adcircResultsComparison:
+	$(MAKE) BUILDTYPE=adcircResultsComparison $@
    hstime:
 	$(MAKE) BUILDTYPE=hstime $@
    aswip:
@@ -522,6 +537,7 @@ else
    buildstwave23:: $(O_DIR)
    hot2asc:: $(O_DIR)
    inflate:: $(O_DIR)
+   adcircResultsComparison:: $(O_DIR)
    hstime::   $(O_DIR)
    aswip::  $(O_DIR)
    adccmp:  $(O_DIR)
@@ -602,6 +618,9 @@ endif
    inflate ::  $(INF_OBJ)
 	$(CF) $(FFLAGS) -o $@ $(wildcard $(O_DIR)*.o)
 
+   adcircResultsComparison :: $(RCMP_OBJ)
+	$(CF) $(FFLAGS) -o $@ $(wildcard $(O_DIR)*.o) $(LIBS) $(LDFLAGS)
+
    hstime ::  $(HST_OBJ)
 	$(CF) $(FFLAGS) -o $@ $(wildcard $(O_DIR)*.o) $(LIBS) $(LDFLAGS)
 
@@ -622,7 +641,7 @@ clean:
 clobber:
 	rm -r -f odir*
 	rm -f adcprep adcprep_be adcirc $(PARALLEL_ADCIRC) adcswan padcswan p15 owi22 build13 aswip \
-           build12 buildstwave23 hot2asc inflate hstime sizes.o adcpost adccmp libadc.a
+           build12 buildstwave23 hot2asc inflate hstime sizes.o adcpost adccmp libadc.a adcircResultsComparison
 	rm -f  ../swan/*.f ../swan/*.for ../swan/*.f90 .swanlastcompile
 help:
 	@echo "This makefile supports the following:"
@@ -639,6 +658,7 @@ help:
 	@echo "make buildstwave23 - makes the fort.23 file generator"
 	@echo "make hot2asc       - makes the hotstart debug program"
 	@echo "make inflate       - makes the output file inflation program"
+	@echo "make adcircResultsComparison    - makes the utility used by the ADCIRC test suite"
 	@echo "make hstime   - makes the hot start time examiner"
 	@echo "make adccmp     - makes the output file comparison program"
 #Casey 091013: Add text for the SWAN+ADCIRC targets.
@@ -799,6 +819,10 @@ $(O_DIR)hot2asc.o     : hot2asc.F
 
 $(O_DIR)inflate.o     : inflate.F
 
+# test suite results compare utility
+
+$(O_DIR)adcircResultsComparison.o      : adcircResultsComparison.F90
+
 # hot start time examiner
 
 $(O_DIR)hstime.o     : hstime.F
